<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Data | Guineafowl-lcWGS Notebook</title>
  <meta name="description" content="A diary-style notebook for this project" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Data | Guineafowl-lcWGS Notebook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A diary-style notebook for this project" />
  <meta name="github-repo" content="https://github.com/GiselaHKopp/Guineafowl-lcWGS" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Data | Guineafowl-lcWGS Notebook" />
  
  <meta name="twitter:description" content="A diary-style notebook for this project" />
  

<meta name="author" content="Gisela H. Kopp" />


<meta name="date" content="2021-09-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="project-notes.html"/>
<link rel="next" href="analysis.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />










<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });
  
  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    // var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeText = $('<img style="width:45px;" src="img/both.svg">');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    var buttonColK = $('<div class="check"><input type="checkbox" id="cc_' + id + '" class="btn btn-default btn-xs code-folding-btn pull-right ch1" /><label class="checktrue" for="cc_' + id + '" ></label></div>');

    buttonColK
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    //buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);
    buttonCol.append(buttonColK);

    div.before(buttonRow);

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text();
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text();
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<script>
$(document).ready(function() {

  // Section anchors
  $('.section h1, .section h2, .section h3, .section h4, .section h5').each(function() {
    anchor = '#' + $(this).parent().attr('id');
    $(this).addClass("hasAnchor").append('<a href="' + anchor + '" class="anchor"></a>');
  });
});
</script>
<link rel="shortcut icon" href="img/favicon.ico" />
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>document.write('<div class="btn-group pull-right" style="position: absolute; top: 35px; right: 2%; z-index: 10"><button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true"aria-expanded="true" data-_extension-text-contrast=""><span>global code</span><span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All</a></li><li><a id="rmd-hide-all-code" href="#">Hide All</a></li></ul></div>')</script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://giselakopp.netlify.app/">Sociality & Evolution</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#dependencies"><i class="fa fa-check"></i><b>1.1</b> Dependencies</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="project-development.html"><a href="project-development.html"><i class="fa fa-check"></i><b>2</b> Project development</a><ul>
<li class="chapter" data-level="2.1" data-path="project-development.html"><a href="project-development.html#general-idea"><i class="fa fa-check"></i><b>2.1</b> General idea</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="project-notes.html"><a href="project-notes.html"><i class="fa fa-check"></i><b>3</b> Project notes</a><ul>
<li class="chapter" data-level="3.1" data-path="project-notes.html"><a href="project-notes.html#reading-notes"><i class="fa fa-check"></i><b>3.1</b> Reading (notes)</a></li>
<li class="chapter" data-level="3.2" data-path="project-notes.html"><a href="project-notes.html#ideas-to-follow-up-on"><i class="fa fa-check"></i><b>3.2</b> Ideas to follow up on</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>4</b> Data</a><ul>
<li class="chapter" data-level="4.1" data-path="data.html"><a href="data.html#setup"><i class="fa fa-check"></i><b>4.1</b> Setup</a></li>
<li class="chapter" data-level="4.2" data-path="data.html"><a href="data.html#data-collection"><i class="fa fa-check"></i><b>4.2</b> Data collection</a></li>
<li class="chapter" data-level="4.3" data-path="data.html"><a href="data.html#data-processing"><i class="fa fa-check"></i><b>4.3</b> Data processing</a><ul>
<li class="chapter" data-level="4.3.1" data-path="data.html"><a href="data.html#sequencing-data"><i class="fa fa-check"></i><b>4.3.1</b> Sequencing data</a></li>
<li class="chapter" data-level="4.3.2" data-path="data.html"><a href="data.html#relatedness-data"><i class="fa fa-check"></i><b>4.3.2</b> Relatedness data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="analysis.html"><a href="analysis.html"><i class="fa fa-check"></i><b>5</b> Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="analysis.html"><a href="analysis.html#setup-1"><i class="fa fa-check"></i><b>5.1</b> Setup</a></li>
<li class="chapter" data-level="5.2" data-path="analysis.html"><a href="analysis.html#data-overview"><i class="fa fa-check"></i><b>5.2</b> Data overview</a></li>
<li class="chapter" data-level="5.3" data-path="analysis.html"><a href="analysis.html#analysis-2"><i class="fa fa-check"></i><b>5.3</b> Analysis 2</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="references-and-session-info.html"><a href="references-and-session-info.html"><i class="fa fa-check"></i><b>6</b> References and Session Info</a><ul>
<li class="chapter" data-level="6.1" data-path="references-and-session-info.html"><a href="references-and-session-info.html#session-info"><i class="fa fa-check"></i><b>6.1</b> Session Info</a></li>
<li class="chapter" data-level="6.2" data-path="references-and-session-info.html"><a href="references-and-session-info.html#references"><i class="fa fa-check"></i><b>6.2</b> References</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Guineafowl-lcWGS Notebook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data" class="section level1">
<h1><span class="header-section-number">4</span> Data</h1>
<p>Data compiled and used in this project</p>
<div id="setup" class="section level2">
<h2><span class="header-section-number">4.1</span> Setup</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb2-2" title="2"><span class="co">#&gt; -- Attaching packages ---------------------------------------------------------------------- tidyverse 1.3.0 --</span></a>
<a class="sourceLine" id="cb2-3" title="3"><span class="co">#&gt; v ggplot2 3.3.2     v purrr   0.3.4</span></a>
<a class="sourceLine" id="cb2-4" title="4"><span class="co">#&gt; v tibble  3.0.3     v dplyr   1.0.2</span></a>
<a class="sourceLine" id="cb2-5" title="5"><span class="co">#&gt; v tidyr   1.1.2     v stringr 1.4.0</span></a>
<a class="sourceLine" id="cb2-6" title="6"><span class="co">#&gt; v readr   1.3.1     v forcats 0.5.0</span></a>
<a class="sourceLine" id="cb2-7" title="7"><span class="co">#&gt; -- Conflicts ------------------------------------------------------------------------- tidyverse_conflicts() --</span></a>
<a class="sourceLine" id="cb2-8" title="8"><span class="co">#&gt; x dplyr::filter() masks stats::filter()</span></a>
<a class="sourceLine" id="cb2-9" title="9"><span class="co">#&gt; x dplyr::lag()    masks stats::lag()</span></a>
<a class="sourceLine" id="cb2-10" title="10"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb2-11" title="11"><span class="kw">library</span>(ggthemes)</a>
<a class="sourceLine" id="cb2-12" title="12"><span class="kw">library</span>(ggforce)</a>
<a class="sourceLine" id="cb2-13" title="13"><span class="kw">library</span>(ggridges)</a>
<a class="sourceLine" id="cb2-14" title="14"><span class="kw">library</span>(ggsci)</a>
<a class="sourceLine" id="cb2-15" title="15"><span class="kw">library</span>(wesanderson)</a>
<a class="sourceLine" id="cb2-16" title="16"><span class="kw">library</span>(ggallin)</a></code></pre></div>
</div>
<div id="data-collection" class="section level2">
<h2><span class="header-section-number">4.2</span> Data collection</h2>
<p>Group information is stored in <em>sequenced_samples_groups.csv</em>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1">sampleID_group &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;vignettes/data/sequenced_samples_groups.csv&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>, <span class="dt">as.is =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-2" title="2"></a>
<a class="sourceLine" id="cb3-3" title="3"></a>
<a class="sourceLine" id="cb3-4" title="4"><span class="kw">str</span>(sampleID_group)</a>
<a class="sourceLine" id="cb3-5" title="5"><span class="co">#&gt; &#39;data.frame&#39;:    95 obs. of  4 variables:</span></a>
<a class="sourceLine" id="cb3-6" title="6"><span class="co">#&gt;  $ Group_ID       : chr  &quot;wing_tags&quot; &quot;wing_tags&quot; &quot;wing_tags&quot; &quot;wing_tags&quot; ...</span></a>
<a class="sourceLine" id="cb3-7" title="7"><span class="co">#&gt;  $ Sample.ind.code: chr  &quot;A1295&quot; &quot;A1296&quot; &quot;A1297&quot; &quot;A1298&quot; ...</span></a>
<a class="sourceLine" id="cb3-8" title="8"><span class="co">#&gt;  $ Real.ind.code  : chr  &quot;W1295&quot; &quot;W1296&quot; &quot;W1297&quot; &quot;W1298&quot; ...</span></a>
<a class="sourceLine" id="cb3-9" title="9"><span class="co">#&gt;  $ Colour_.bands  : chr  &quot;WT002&quot; &quot;WT003&quot; &quot;WT004&quot; &quot;WT021&quot; ...</span></a>
<a class="sourceLine" id="cb3-10" title="10"></a>
<a class="sourceLine" id="cb3-11" title="11"><span class="co">#rename Real.ind.code to ID</span></a>
<a class="sourceLine" id="cb3-12" title="12">sampleID_group &lt;-<span class="st"> </span><span class="kw">rename</span>(sampleID_group, <span class="st">&quot;ID&quot;</span> =<span class="st"> &quot;Sample.ind.code&quot;</span>)</a>
<a class="sourceLine" id="cb3-13" title="13"></a>
<a class="sourceLine" id="cb3-14" title="14"><span class="co">#only keep group and ID</span></a>
<a class="sourceLine" id="cb3-15" title="15">sampleID_group &lt;-<span class="st"> </span><span class="kw">select</span>(sampleID_group, Group_ID, ID)</a>
<a class="sourceLine" id="cb3-16" title="16"><span class="kw">str</span>(sampleID_group)</a>
<a class="sourceLine" id="cb3-17" title="17"><span class="co">#&gt; &#39;data.frame&#39;:    95 obs. of  2 variables:</span></a>
<a class="sourceLine" id="cb3-18" title="18"><span class="co">#&gt;  $ Group_ID: chr  &quot;wing_tags&quot; &quot;wing_tags&quot; &quot;wing_tags&quot; &quot;wing_tags&quot; ...</span></a>
<a class="sourceLine" id="cb3-19" title="19"><span class="co">#&gt;  $ ID      : chr  &quot;A1295&quot; &quot;A1296&quot; &quot;A1297&quot; &quot;A1298&quot; ...</span></a></code></pre></div>
<p>There are three samples that are not listed correctly and would lead to NA’s in downstream analysis, so we manually adjust them here</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="co">#correct sample B1299 is missing</span></a>
<a class="sourceLine" id="cb4-2" title="2">sample_info_add &lt;-<span class="st"> </span><span class="kw">data.frame</span> (<span class="dt">Group_ID =</span> <span class="kw">c</span>(<span class="st">&quot;wing_tags&quot;</span>, <span class="st">&quot;wing_tags&quot;</span>, <span class="st">&quot;5509&quot;</span>),</a>
<a class="sourceLine" id="cb4-3" title="3">                               <span class="dt">ID =</span> <span class="kw">c</span>(<span class="st">&quot;B1299&quot;</span>, <span class="st">&quot;W1304&quot;</span>, <span class="st">&quot;W1345&quot;</span>))</a>
<a class="sourceLine" id="cb4-4" title="4">sample_info_add</a>
<a class="sourceLine" id="cb4-5" title="5"><span class="co">#&gt;    Group_ID    ID</span></a>
<a class="sourceLine" id="cb4-6" title="6"><span class="co">#&gt; 1 wing_tags B1299</span></a>
<a class="sourceLine" id="cb4-7" title="7"><span class="co">#&gt; 2 wing_tags W1304</span></a>
<a class="sourceLine" id="cb4-8" title="8"><span class="co">#&gt; 3      5509 W1345</span></a>
<a class="sourceLine" id="cb4-9" title="9"></a>
<a class="sourceLine" id="cb4-10" title="10">sampleID_group &lt;-<span class="st"> </span><span class="kw">rbind</span>(sampleID_group, sample_info_add)</a></code></pre></div>
<p>Sex information is stored in <em>samples_sex.csv</em>. Sex was determined in the field and confirmed at least once with a molecular sexing method in the lab. For cases with mismatch between field and molecular sex we set the sex to undetermined.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1">sampleID_sex &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;vignettes/data/samples_sex.csv&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;;&quot;</span>, <span class="dt">as.is =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb5-2" title="2"><span class="kw">str</span>(sampleID_sex)</a>
<a class="sourceLine" id="cb5-3" title="3"><span class="co">#&gt; &#39;data.frame&#39;:    133 obs. of  2 variables:</span></a>
<a class="sourceLine" id="cb5-4" title="4"><span class="co">#&gt;  $ ID : chr  &quot;A1295&quot; &quot;A1296&quot; &quot;A1297&quot; &quot;A1298&quot; ...</span></a>
<a class="sourceLine" id="cb5-5" title="5"><span class="co">#&gt;  $ sex: chr  &quot;m&quot; &quot;f&quot; &quot;m&quot; &quot;m&quot; ...</span></a></code></pre></div>
<p>merge group and sex information</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1"></a>
<a class="sourceLine" id="cb6-2" title="2">sample_info &lt;-<span class="st"> </span><span class="kw">full_join</span>(sampleID_group, sampleID_sex, <span class="dt">by =</span> <span class="st">&quot;ID&quot;</span>)</a>
<a class="sourceLine" id="cb6-3" title="3"><span class="kw">str</span>(sample_info)</a>
<a class="sourceLine" id="cb6-4" title="4"><span class="co">#&gt; &#39;data.frame&#39;:    133 obs. of  3 variables:</span></a>
<a class="sourceLine" id="cb6-5" title="5"><span class="co">#&gt;  $ Group_ID: chr  &quot;wing_tags&quot; &quot;wing_tags&quot; &quot;wing_tags&quot; &quot;wing_tags&quot; ...</span></a>
<a class="sourceLine" id="cb6-6" title="6"><span class="co">#&gt;  $ ID      : chr  &quot;A1295&quot; &quot;A1296&quot; &quot;A1297&quot; &quot;A1298&quot; ...</span></a>
<a class="sourceLine" id="cb6-7" title="7"><span class="co">#&gt;  $ sex     : chr  &quot;m&quot; &quot;f&quot; &quot;m&quot; &quot;m&quot; ...</span></a></code></pre></div>
</div>
<div id="data-processing" class="section level2">
<h2><span class="header-section-number">4.3</span> Data processing</h2>
<div id="sequencing-data" class="section level3">
<h3><span class="header-section-number">4.3.1</span> Sequencing data</h3>
<p>First, we load an overview of the analysed sequencing libraries. They are stored in the folder <code>pipeline</code> as <em>file_list.tsv</em>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">seqlib_overview &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;vignettes/data/pipeline/file_list.tsv&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>,  <span class="dt">as.is =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb7-2" title="2"><span class="kw">str</span>(seqlib_overview)</a>
<a class="sourceLine" id="cb7-3" title="3"><span class="co">#&gt; &#39;data.frame&#39;:    238 obs. of  6 variables:</span></a>
<a class="sourceLine" id="cb7-4" title="4"><span class="co">#&gt;  $ file_name: chr  &quot;mpg_L16936-1_W1350_S1_R1_001.fastq.gz&quot; &quot;mpg_L16936-1_W1350_S1_R2_001.fastq.gz&quot; &quot;mpg_L16926-1_W1398_S2_R1_001.fastq.gz&quot; &quot;mpg_L16926-1_W1398_S2_R2_001.fastq.gz&quot; ...</span></a>
<a class="sourceLine" id="cb7-5" title="5"><span class="co">#&gt;  $ RGID     : int  1 1 1 1 1 1 1 1 1 1 ...</span></a>
<a class="sourceLine" id="cb7-6" title="6"><span class="co">#&gt;  $ RGLB     : chr  &quot;lib1&quot; &quot;lib1&quot; &quot;lib1&quot; &quot;lib1&quot; ...</span></a>
<a class="sourceLine" id="cb7-7" title="7"><span class="co">#&gt;  $ RGPL     : chr  &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; ...</span></a>
<a class="sourceLine" id="cb7-8" title="8"><span class="co">#&gt;  $ RGPU     : chr  &quot;unit1&quot; &quot;unit1&quot; &quot;unit1&quot; &quot;unit1&quot; ...</span></a>
<a class="sourceLine" id="cb7-9" title="9"><span class="co">#&gt;  $ RGSM     : chr  &quot;RGID1_S1&quot; &quot;RGID1_S1&quot; &quot;RGID1_S2&quot; &quot;RGID1_S2&quot; ...</span></a></code></pre></div>
<p>The information about the sampled individual, which we will later need to assign the relatedness estimation to specific individuals, is available in the <em>file_name</em> column. We will extract this information and put it in an extra <em>ID</em> column.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1"><span class="co">#first need to create a new column that includes the sample ID</span></a>
<a class="sourceLine" id="cb8-2" title="2">seqlib_overview<span class="op">$</span>ID &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;([[:alnum:]]{3}_[[:alnum:]]{6}-[[:digit:]])_([[:alnum:]]{5,7})_([[:alnum:]]{2,3}_[[:alnum:]]{2}_[[:graph:]]{12})&quot;</span>, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">2&quot;</span>, seqlib_overview<span class="op">$</span>file_name)</a>
<a class="sourceLine" id="cb8-3" title="3"></a>
<a class="sourceLine" id="cb8-4" title="4"><span class="kw">str</span>(seqlib_overview)</a>
<a class="sourceLine" id="cb8-5" title="5"><span class="co">#&gt; &#39;data.frame&#39;:    238 obs. of  7 variables:</span></a>
<a class="sourceLine" id="cb8-6" title="6"><span class="co">#&gt;  $ file_name: chr  &quot;mpg_L16936-1_W1350_S1_R1_001.fastq.gz&quot; &quot;mpg_L16936-1_W1350_S1_R2_001.fastq.gz&quot; &quot;mpg_L16926-1_W1398_S2_R1_001.fastq.gz&quot; &quot;mpg_L16926-1_W1398_S2_R2_001.fastq.gz&quot; ...</span></a>
<a class="sourceLine" id="cb8-7" title="7"><span class="co">#&gt;  $ RGID     : int  1 1 1 1 1 1 1 1 1 1 ...</span></a>
<a class="sourceLine" id="cb8-8" title="8"><span class="co">#&gt;  $ RGLB     : chr  &quot;lib1&quot; &quot;lib1&quot; &quot;lib1&quot; &quot;lib1&quot; ...</span></a>
<a class="sourceLine" id="cb8-9" title="9"><span class="co">#&gt;  $ RGPL     : chr  &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; ...</span></a>
<a class="sourceLine" id="cb8-10" title="10"><span class="co">#&gt;  $ RGPU     : chr  &quot;unit1&quot; &quot;unit1&quot; &quot;unit1&quot; &quot;unit1&quot; ...</span></a>
<a class="sourceLine" id="cb8-11" title="11"><span class="co">#&gt;  $ RGSM     : chr  &quot;RGID1_S1&quot; &quot;RGID1_S1&quot; &quot;RGID1_S2&quot; &quot;RGID1_S2&quot; ...</span></a>
<a class="sourceLine" id="cb8-12" title="12"><span class="co">#&gt;  $ ID       : chr  &quot;W1350&quot; &quot;W1350&quot; &quot;W1398&quot; &quot;W1398&quot; ...</span></a></code></pre></div>
<p>Next, we evaluate the quality of the data and the different sequencing approaches.
An overview of the data as output from the Nextflow pipeline as of 29th July 2021 is stored in <em>preprocessing_overview.tsv</em></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">data_overview &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;vignettes/data/pipeline/preprocessing_overview.tsv&quot;</span>, <span class="dt">as.is =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-2" title="2"><span class="kw">str</span>(data_overview)</a>
<a class="sourceLine" id="cb9-3" title="3"><span class="co">#&gt; &#39;data.frame&#39;:    119 obs. of  31 variables:</span></a>
<a class="sourceLine" id="cb9-4" title="4"><span class="co">#&gt;  $ RGSM                                     : chr  &quot;RGID1_S1&quot; &quot;RGID1_S2&quot; &quot;RGID1_S3&quot; &quot;RGID1_S4&quot; ...</span></a>
<a class="sourceLine" id="cb9-5" title="5"><span class="co">#&gt;  $ RGID                                     : int  1 1 1 1 1 1 1 1 1 1 ...</span></a>
<a class="sourceLine" id="cb9-6" title="6"><span class="co">#&gt;  $ RGLB                                     : chr  &quot;lib1&quot; &quot;lib1&quot; &quot;lib1&quot; &quot;lib1&quot; ...</span></a>
<a class="sourceLine" id="cb9-7" title="7"><span class="co">#&gt;  $ RGPL                                     : chr  &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; ...</span></a>
<a class="sourceLine" id="cb9-8" title="8"><span class="co">#&gt;  $ RGPU                                     : chr  &quot;unit1&quot; &quot;unit1&quot; &quot;unit1&quot; &quot;unit1&quot; ...</span></a>
<a class="sourceLine" id="cb9-9" title="9"><span class="co">#&gt;  $ filename_one                             : chr  &quot;mpg_L16936-1_W1350_S1_R1_001.fastq.gz&quot; &quot;mpg_L16926-1_W1398_S2_R1_001.fastq.gz&quot; &quot;mpg_L16927-1_W1429_S3_R1_001.fastq.gz&quot; &quot;mpg_L16928-1_W1502_S4_R1_001.fastq.gz&quot; ...</span></a>
<a class="sourceLine" id="cb9-10" title="10"><span class="co">#&gt;  $ filename_two                             : chr  &quot;mpg_L16936-1_W1350_S1_R2_001.fastq.gz&quot; &quot;mpg_L16926-1_W1398_S2_R2_001.fastq.gz&quot; &quot;mpg_L16927-1_W1429_S3_R2_001.fastq.gz&quot; &quot;mpg_L16928-1_W1502_S4_R2_001.fastq.gz&quot; ...</span></a>
<a class="sourceLine" id="cb9-11" title="11"><span class="co">#&gt;  $ reads_pre_trim_one                       : int  3844063 3360132 3034920 6695288 6040662 2842383 3879834 6087445 5686616 4107154 ...</span></a>
<a class="sourceLine" id="cb9-12" title="12"><span class="co">#&gt;  $ reads_pre_trim_two                       : int  3844063 3360132 3034920 6695288 6040662 2842383 3879834 6087445 5686616 4107154 ...</span></a>
<a class="sourceLine" id="cb9-13" title="13"><span class="co">#&gt;  $ reads_pre_trim_total                     : int  7688126 6720264 6069840 13390576 12081324 5684766 7759668 12174890 11373232 8214308 ...</span></a>
<a class="sourceLine" id="cb9-14" title="14"><span class="co">#&gt;  $ reads_post_trim_one                      : int  3334957 3057420 2775271 6144973 5496263 2618095 3544978 5592586 5206646 3740830 ...</span></a>
<a class="sourceLine" id="cb9-15" title="15"><span class="co">#&gt;  $ reads_post_trim_two                      : int  3317567 2622060 2444309 5472149 4763684 2298940 2933104 4916336 4551423 3252219 ...</span></a>
<a class="sourceLine" id="cb9-16" title="16"><span class="co">#&gt;  $ reads_post_trim_total                    : int  6652524 5679480 5219580 11617122 10259947 4917035 6478082 10508922 9758069 6993049 ...</span></a>
<a class="sourceLine" id="cb9-17" title="17"><span class="co">#&gt;  $ reads_trimmed_lost_one                   : int  509106 302712 259649 550315 544399 224288 334856 494859 479970 366324 ...</span></a>
<a class="sourceLine" id="cb9-18" title="18"><span class="co">#&gt;  $ reads_trimmed_lost_two                   : int  526496 738072 590611 1223139 1276978 543443 946730 1171109 1135193 854935 ...</span></a>
<a class="sourceLine" id="cb9-19" title="19"><span class="co">#&gt;  $ reads_trimmed_lost_total                 : int  1035602 1040784 850260 1773454 1821377 767731 1281586 1665968 1615163 1221259 ...</span></a>
<a class="sourceLine" id="cb9-20" title="20"><span class="co">#&gt;  $ reads_trimmed_lost_total_percent         : num  0.135 0.155 0.14 0.132 0.151 ...</span></a>
<a class="sourceLine" id="cb9-21" title="21"><span class="co">#&gt;  $ reads_mapped                             : int  6448222 5897151 5344556 11948516 10542799 5052562 6862719 10880329 10069374 7213292 ...</span></a>
<a class="sourceLine" id="cb9-22" title="22"><span class="co">#&gt;  $ reads_mapped_and_paired                  : int  6124942 4910684 4546908 10259622 8836944 4295176 5524338 9244878 8510126 6063336 ...</span></a>
<a class="sourceLine" id="cb9-23" title="23"><span class="co">#&gt;  $ reads_unmapped                           : int  158628 176787 170192 251546 386867 143020 186719 234815 276382 223486 ...</span></a>
<a class="sourceLine" id="cb9-24" title="24"><span class="co">#&gt;  $ reads_mapped_percent                     : num  0.975 0.97 0.968 0.979 0.963 ...</span></a>
<a class="sourceLine" id="cb9-25" title="25"><span class="co">#&gt;  $ average_coverage                         : num  1.6 1.63 1.58 2.16 2.07 ...</span></a>
<a class="sourceLine" id="cb9-26" title="26"><span class="co">#&gt;  $ average_coverarge_stdev                  : num  2.67 2.8 2.62 2.98 3.64 ...</span></a>
<a class="sourceLine" id="cb9-27" title="27"><span class="co">#&gt;  $ unpaired_reads_examined_for_deduplication: int  323280 986467 797648 1688894 1705855 757386 1338381 1635451 1559248 1149956 ...</span></a>
<a class="sourceLine" id="cb9-28" title="28"><span class="co">#&gt;  $ paired_reads_examined_for_deduplication  : int  3094003 2475793 2291351 5174753 4449902 2167892 2782428 4657453 4288831 3054109 ...</span></a>
<a class="sourceLine" id="cb9-29" title="29"><span class="co">#&gt;  $ unpaired_read_duplicated                 : int  66109 106320 78976 195164 185282 87774 136509 172933 169988 118239 ...</span></a>
<a class="sourceLine" id="cb9-30" title="30"><span class="co">#&gt;  $ paired_read_duplicates                   : int  221442 119369 103005 260460 197926 115543 120104 206962 194659 133521 ...</span></a>
<a class="sourceLine" id="cb9-31" title="31"><span class="co">#&gt;  $ percent_duplication                      : num  0.0782 0.0581 0.053 0.0595 0.0548 ...</span></a>
<a class="sourceLine" id="cb9-32" title="32"><span class="co">#&gt;  $ sequenced_library_complexity             : int  6090385 5720981 5222642 11474365 10330774 4870312 6664743 10519461 9774928 7042018 ...</span></a>
<a class="sourceLine" id="cb9-33" title="33"><span class="co">#&gt;  $ estimated_library_complexity             : int  23660823 29649178 29602264 59328308 57146455 23492191 37280729 60581428 54839066 40354282 ...</span></a>
<a class="sourceLine" id="cb9-34" title="34"><span class="co">#&gt;  $ percent_library_sequenced                : num  0.257 0.193 0.176 0.193 0.181 ...</span></a></code></pre></div>
<p>One sequencing library was resequenced in an additionl run but wrongly assigned to RGID 1, so we correct this</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1"></a>
<a class="sourceLine" id="cb10-2" title="2">data_overview<span class="op">$</span>RGID[data_overview<span class="op">$</span>filename_one <span class="op">==</span><span class="st"> &quot;mpg_L16936-1_W1350_S1_R1_001.fastq.gz&quot;</span>] &lt;-<span class="st"> &quot;2&quot;</span></a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1"><span class="co">#create new column that specifies if sequencing results are from original run or from resequencing</span></a>
<a class="sourceLine" id="cb11-2" title="2">data_overview &lt;-<span class="st"> </span>data_overview <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb11-3" title="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">original_reseq =</span> <span class="kw">case_when</span>(RGID <span class="op">==</span><span class="st"> &quot;1&quot;</span> <span class="op">~</span><span class="st"> &quot;original&quot;</span>,</a>
<a class="sourceLine" id="cb11-4" title="4">                                RGID <span class="op">==</span><span class="st"> &quot;2&quot;</span> <span class="op">~</span><span class="st"> &quot;reseq&quot;</span>))</a></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1"><span class="co">#first need to create a new column that includes the sample ID</span></a>
<a class="sourceLine" id="cb12-2" title="2">data_overview<span class="op">$</span>ID &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;([[:alnum:]]{3}_[[:alnum:]]{6}-[[:digit:]])_([[:alnum:]]{5,7})_([[:alnum:]]{2,3}_[[:alnum:]]{2}_[[:graph:]]{12})&quot;</span>, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">2&quot;</span>, data_overview<span class="op">$</span>filename_one)</a>
<a class="sourceLine" id="cb12-3" title="3"></a>
<a class="sourceLine" id="cb12-4" title="4"><span class="kw">str</span>(data_overview)</a>
<a class="sourceLine" id="cb12-5" title="5"><span class="co">#&gt; &#39;data.frame&#39;:    119 obs. of  33 variables:</span></a>
<a class="sourceLine" id="cb12-6" title="6"><span class="co">#&gt;  $ RGSM                                     : chr  &quot;RGID1_S1&quot; &quot;RGID1_S2&quot; &quot;RGID1_S3&quot; &quot;RGID1_S4&quot; ...</span></a>
<a class="sourceLine" id="cb12-7" title="7"><span class="co">#&gt;  $ RGID                                     : chr  &quot;2&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; ...</span></a>
<a class="sourceLine" id="cb12-8" title="8"><span class="co">#&gt;  $ RGLB                                     : chr  &quot;lib1&quot; &quot;lib1&quot; &quot;lib1&quot; &quot;lib1&quot; ...</span></a>
<a class="sourceLine" id="cb12-9" title="9"><span class="co">#&gt;  $ RGPL                                     : chr  &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; &quot;ILLUMINA&quot; ...</span></a>
<a class="sourceLine" id="cb12-10" title="10"><span class="co">#&gt;  $ RGPU                                     : chr  &quot;unit1&quot; &quot;unit1&quot; &quot;unit1&quot; &quot;unit1&quot; ...</span></a>
<a class="sourceLine" id="cb12-11" title="11"><span class="co">#&gt;  $ filename_one                             : chr  &quot;mpg_L16936-1_W1350_S1_R1_001.fastq.gz&quot; &quot;mpg_L16926-1_W1398_S2_R1_001.fastq.gz&quot; &quot;mpg_L16927-1_W1429_S3_R1_001.fastq.gz&quot; &quot;mpg_L16928-1_W1502_S4_R1_001.fastq.gz&quot; ...</span></a>
<a class="sourceLine" id="cb12-12" title="12"><span class="co">#&gt;  $ filename_two                             : chr  &quot;mpg_L16936-1_W1350_S1_R2_001.fastq.gz&quot; &quot;mpg_L16926-1_W1398_S2_R2_001.fastq.gz&quot; &quot;mpg_L16927-1_W1429_S3_R2_001.fastq.gz&quot; &quot;mpg_L16928-1_W1502_S4_R2_001.fastq.gz&quot; ...</span></a>
<a class="sourceLine" id="cb12-13" title="13"><span class="co">#&gt;  $ reads_pre_trim_one                       : int  3844063 3360132 3034920 6695288 6040662 2842383 3879834 6087445 5686616 4107154 ...</span></a>
<a class="sourceLine" id="cb12-14" title="14"><span class="co">#&gt;  $ reads_pre_trim_two                       : int  3844063 3360132 3034920 6695288 6040662 2842383 3879834 6087445 5686616 4107154 ...</span></a>
<a class="sourceLine" id="cb12-15" title="15"><span class="co">#&gt;  $ reads_pre_trim_total                     : int  7688126 6720264 6069840 13390576 12081324 5684766 7759668 12174890 11373232 8214308 ...</span></a>
<a class="sourceLine" id="cb12-16" title="16"><span class="co">#&gt;  $ reads_post_trim_one                      : int  3334957 3057420 2775271 6144973 5496263 2618095 3544978 5592586 5206646 3740830 ...</span></a>
<a class="sourceLine" id="cb12-17" title="17"><span class="co">#&gt;  $ reads_post_trim_two                      : int  3317567 2622060 2444309 5472149 4763684 2298940 2933104 4916336 4551423 3252219 ...</span></a>
<a class="sourceLine" id="cb12-18" title="18"><span class="co">#&gt;  $ reads_post_trim_total                    : int  6652524 5679480 5219580 11617122 10259947 4917035 6478082 10508922 9758069 6993049 ...</span></a>
<a class="sourceLine" id="cb12-19" title="19"><span class="co">#&gt;  $ reads_trimmed_lost_one                   : int  509106 302712 259649 550315 544399 224288 334856 494859 479970 366324 ...</span></a>
<a class="sourceLine" id="cb12-20" title="20"><span class="co">#&gt;  $ reads_trimmed_lost_two                   : int  526496 738072 590611 1223139 1276978 543443 946730 1171109 1135193 854935 ...</span></a>
<a class="sourceLine" id="cb12-21" title="21"><span class="co">#&gt;  $ reads_trimmed_lost_total                 : int  1035602 1040784 850260 1773454 1821377 767731 1281586 1665968 1615163 1221259 ...</span></a>
<a class="sourceLine" id="cb12-22" title="22"><span class="co">#&gt;  $ reads_trimmed_lost_total_percent         : num  0.135 0.155 0.14 0.132 0.151 ...</span></a>
<a class="sourceLine" id="cb12-23" title="23"><span class="co">#&gt;  $ reads_mapped                             : int  6448222 5897151 5344556 11948516 10542799 5052562 6862719 10880329 10069374 7213292 ...</span></a>
<a class="sourceLine" id="cb12-24" title="24"><span class="co">#&gt;  $ reads_mapped_and_paired                  : int  6124942 4910684 4546908 10259622 8836944 4295176 5524338 9244878 8510126 6063336 ...</span></a>
<a class="sourceLine" id="cb12-25" title="25"><span class="co">#&gt;  $ reads_unmapped                           : int  158628 176787 170192 251546 386867 143020 186719 234815 276382 223486 ...</span></a>
<a class="sourceLine" id="cb12-26" title="26"><span class="co">#&gt;  $ reads_mapped_percent                     : num  0.975 0.97 0.968 0.979 0.963 ...</span></a>
<a class="sourceLine" id="cb12-27" title="27"><span class="co">#&gt;  $ average_coverage                         : num  1.6 1.63 1.58 2.16 2.07 ...</span></a>
<a class="sourceLine" id="cb12-28" title="28"><span class="co">#&gt;  $ average_coverarge_stdev                  : num  2.67 2.8 2.62 2.98 3.64 ...</span></a>
<a class="sourceLine" id="cb12-29" title="29"><span class="co">#&gt;  $ unpaired_reads_examined_for_deduplication: int  323280 986467 797648 1688894 1705855 757386 1338381 1635451 1559248 1149956 ...</span></a>
<a class="sourceLine" id="cb12-30" title="30"><span class="co">#&gt;  $ paired_reads_examined_for_deduplication  : int  3094003 2475793 2291351 5174753 4449902 2167892 2782428 4657453 4288831 3054109 ...</span></a>
<a class="sourceLine" id="cb12-31" title="31"><span class="co">#&gt;  $ unpaired_read_duplicated                 : int  66109 106320 78976 195164 185282 87774 136509 172933 169988 118239 ...</span></a>
<a class="sourceLine" id="cb12-32" title="32"><span class="co">#&gt;  $ paired_read_duplicates                   : int  221442 119369 103005 260460 197926 115543 120104 206962 194659 133521 ...</span></a>
<a class="sourceLine" id="cb12-33" title="33"><span class="co">#&gt;  $ percent_duplication                      : num  0.0782 0.0581 0.053 0.0595 0.0548 ...</span></a>
<a class="sourceLine" id="cb12-34" title="34"><span class="co">#&gt;  $ sequenced_library_complexity             : int  6090385 5720981 5222642 11474365 10330774 4870312 6664743 10519461 9774928 7042018 ...</span></a>
<a class="sourceLine" id="cb12-35" title="35"><span class="co">#&gt;  $ estimated_library_complexity             : int  23660823 29649178 29602264 59328308 57146455 23492191 37280729 60581428 54839066 40354282 ...</span></a>
<a class="sourceLine" id="cb12-36" title="36"><span class="co">#&gt;  $ percent_library_sequenced                : num  0.257 0.193 0.176 0.193 0.181 ...</span></a>
<a class="sourceLine" id="cb12-37" title="37"><span class="co">#&gt;  $ original_reseq                           : chr  &quot;reseq&quot; &quot;original&quot; &quot;original&quot; &quot;original&quot; ...</span></a>
<a class="sourceLine" id="cb12-38" title="38"><span class="co">#&gt;  $ ID                                       : chr  &quot;W1350&quot; &quot;W1398&quot; &quot;W1429&quot; &quot;W1502&quot; ...</span></a></code></pre></div>
<p>save data overview</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1"><span class="kw">save</span>(data_overview, <span class="dt">file =</span> <span class="st">&quot;data_overview.RData&quot;</span>)</a></code></pre></div>
<p>Calculate total and mean number of raw reads
!!paired-end sequencing, so need to account for that</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1"></a>
<a class="sourceLine" id="cb14-2" title="2">total_raw_reads &lt;-<span class="st"> </span><span class="kw">sum</span>(data_overview<span class="op">$</span>reads_pre_trim_total)</a>
<a class="sourceLine" id="cb14-3" title="3">total_raw_reads</a>
<a class="sourceLine" id="cb14-4" title="4"><span class="co">#&gt; [1] 1393934732</span></a>
<a class="sourceLine" id="cb14-5" title="5"></a>
<a class="sourceLine" id="cb14-6" title="6">mean_raw_reads &lt;-<span class="st"> </span><span class="kw">mean</span>(data_overview<span class="op">$</span>reads_pre_trim_total)</a>
<a class="sourceLine" id="cb14-7" title="7">mean_raw_reads</a>
<a class="sourceLine" id="cb14-8" title="8"><span class="co">#&gt; [1] 11713737</span></a>
<a class="sourceLine" id="cb14-9" title="9"></a>
<a class="sourceLine" id="cb14-10" title="10">sd_raw_reads &lt;-<span class="st"> </span><span class="kw">sd</span>(data_overview<span class="op">$</span>reads_pre_trim_total)</a>
<a class="sourceLine" id="cb14-11" title="11">sd_raw_reads</a>
<a class="sourceLine" id="cb14-12" title="12"><span class="co">#&gt; [1] 5979400</span></a></code></pre></div>
<p>Calculate total and mean number of quality filtered reads</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1"></a>
<a class="sourceLine" id="cb15-2" title="2">total_filtered_reads &lt;-<span class="st"> </span><span class="kw">sum</span>(data_overview<span class="op">$</span>reads_post_trim_total)</a>
<a class="sourceLine" id="cb15-3" title="3">total_filtered_reads</a>
<a class="sourceLine" id="cb15-4" title="4"><span class="co">#&gt; [1] 1218247416</span></a>
<a class="sourceLine" id="cb15-5" title="5"></a>
<a class="sourceLine" id="cb15-6" title="6">percent_filtered_reads &lt;-<span class="st"> </span>total_filtered_reads<span class="op">/</span>total_raw_reads  </a>
<a class="sourceLine" id="cb15-7" title="7">percent_filtered_reads</a>
<a class="sourceLine" id="cb15-8" title="8"><span class="co">#&gt; [1] 0.873963</span></a>
<a class="sourceLine" id="cb15-9" title="9"></a>
<a class="sourceLine" id="cb15-10" title="10">mean_filtered_reads &lt;-<span class="st"> </span><span class="kw">mean</span>(data_overview<span class="op">$</span>reads_post_trim_total)</a>
<a class="sourceLine" id="cb15-11" title="11">mean_filtered_reads</a>
<a class="sourceLine" id="cb15-12" title="12"><span class="co">#&gt; [1] 10237373</span></a>
<a class="sourceLine" id="cb15-13" title="13"></a>
<a class="sourceLine" id="cb15-14" title="14">percent_mean_filtered_reads &lt;-<span class="st"> </span>mean_filtered_reads<span class="op">/</span>mean_raw_reads</a>
<a class="sourceLine" id="cb15-15" title="15">percent_mean_filtered_reads</a>
<a class="sourceLine" id="cb15-16" title="16"><span class="co">#&gt; [1] 0.873963</span></a>
<a class="sourceLine" id="cb15-17" title="17"></a>
<a class="sourceLine" id="cb15-18" title="18">sd_filtered_reads &lt;-<span class="st"> </span><span class="kw">sd</span>(data_overview<span class="op">$</span>reads_post_trim_total)</a>
<a class="sourceLine" id="cb15-19" title="19">sd_filtered_reads</a>
<a class="sourceLine" id="cb15-20" title="20"><span class="co">#&gt; [1] 5426886</span></a>
<a class="sourceLine" id="cb15-21" title="21"></a>
<a class="sourceLine" id="cb15-22" title="22"></a>
<a class="sourceLine" id="cb15-23" title="23">median_Trimmomatic_dropped &lt;-<span class="st"> </span><span class="kw">median</span>(data_overview<span class="op">$</span>reads_trimmed_lost_total, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb15-24" title="24">median_Trimmomatic_dropped</a>
<a class="sourceLine" id="cb15-25" title="25"><span class="co">#&gt; [1] 1310301</span></a>
<a class="sourceLine" id="cb15-26" title="26"></a>
<a class="sourceLine" id="cb15-27" title="27">IQR_Trimmomatic_dropped &lt;-<span class="st"> </span><span class="kw">IQR</span>(data_overview<span class="op">$</span>reads_trimmed_lost_total, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb15-28" title="28">IQR_Trimmomatic_dropped</a>
<a class="sourceLine" id="cb15-29" title="29"><span class="co">#&gt; [1] 749127</span></a>
<a class="sourceLine" id="cb15-30" title="30"></a>
<a class="sourceLine" id="cb15-31" title="31">range_Trimmomatic_dropped &lt;-<span class="st"> </span><span class="kw">range</span>(data_overview<span class="op">$</span>reads_trimmed_lost_total, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb15-32" title="32">range_Trimmomatic_dropped</a>
<a class="sourceLine" id="cb15-33" title="33"><span class="co">#&gt; [1]  323232 4158669</span></a></code></pre></div>
<p>Calculate mapped reads</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" title="1">median_mapped_reads &lt;-<span class="st"> </span><span class="kw">median</span>(data_overview<span class="op">$</span>reads_mapped)</a>
<a class="sourceLine" id="cb16-2" title="2">median_mapped_reads</a>
<a class="sourceLine" id="cb16-3" title="3"><span class="co">#&gt; [1] 9498515</span></a>
<a class="sourceLine" id="cb16-4" title="4"></a>
<a class="sourceLine" id="cb16-5" title="5">IQR_mapped_reads &lt;-<span class="st"> </span><span class="kw">IQR</span>(data_overview<span class="op">$</span>reads_mapped)</a>
<a class="sourceLine" id="cb16-6" title="6">IQR_mapped_reads</a>
<a class="sourceLine" id="cb16-7" title="7"><span class="co">#&gt; [1] 6902176</span></a>
<a class="sourceLine" id="cb16-8" title="8"></a>
<a class="sourceLine" id="cb16-9" title="9">range_mapped_reads &lt;-<span class="st"> </span><span class="kw">range</span>(data_overview<span class="op">$</span>reads_mapped)</a>
<a class="sourceLine" id="cb16-10" title="10">range_mapped_reads</a>
<a class="sourceLine" id="cb16-11" title="11"><span class="co">#&gt; [1]  1843218 32346499</span></a></code></pre></div>
<p>Calculate duplicates</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1">median_duplicates &lt;-<span class="st"> </span><span class="kw">median</span>(data_overview<span class="op">$</span>percent_duplication)</a>
<a class="sourceLine" id="cb17-2" title="2">median_duplicates</a>
<a class="sourceLine" id="cb17-3" title="3"><span class="co">#&gt; [1] 0.056526</span></a>
<a class="sourceLine" id="cb17-4" title="4"></a>
<a class="sourceLine" id="cb17-5" title="5">range_duplicates &lt;-<span class="st"> </span><span class="kw">range</span>(data_overview<span class="op">$</span>percent_duplication)</a>
<a class="sourceLine" id="cb17-6" title="6">range_duplicates</a>
<a class="sourceLine" id="cb17-7" title="7"><span class="co">#&gt; [1] 0.039762 0.100763</span></a></code></pre></div>
<p>Calculate final coverage</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" title="1">median_final_coverage &lt;-<span class="st"> </span><span class="kw">median</span>(data_overview<span class="op">$</span>average_coverage)</a>
<a class="sourceLine" id="cb18-2" title="2">median_final_coverage</a>
<a class="sourceLine" id="cb18-3" title="3"><span class="co">#&gt; [1] 1.91596</span></a>
<a class="sourceLine" id="cb18-4" title="4"></a>
<a class="sourceLine" id="cb18-5" title="5">IQR_final_coverage &lt;-<span class="st"> </span><span class="kw">IQR</span>(data_overview<span class="op">$</span>average_coverage)</a>
<a class="sourceLine" id="cb18-6" title="6">IQR_final_coverage</a>
<a class="sourceLine" id="cb18-7" title="7"><span class="co">#&gt; [1] 0.601805</span></a>
<a class="sourceLine" id="cb18-8" title="8"></a>
<a class="sourceLine" id="cb18-9" title="9">range_final_coverage &lt;-<span class="st"> </span><span class="kw">range</span>(data_overview<span class="op">$</span>average_coverage)</a>
<a class="sourceLine" id="cb18-10" title="10">range_final_coverage</a>
<a class="sourceLine" id="cb18-11" title="11"><span class="co">#&gt; [1] 1.33748 4.31431</span></a></code></pre></div>
<p>For the sequencing success, we compare the reads that passed Quality control (FastCQ and Trimmotaic) depending on original run (RGID = 1) and resequencing (RGID = 2).</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_bw</span>()) <span class="co">#set theme to black and white</span></a>
<a class="sourceLine" id="cb19-2" title="2"></a>
<a class="sourceLine" id="cb19-3" title="3"><span class="co">#compare reads (y-axis) between original vs. resequenced (x-axis, sorted to have original first) and color according to library prep (shotgun vs. capture)</span></a>
<a class="sourceLine" id="cb19-4" title="4"><span class="kw">ggplot</span>(data_overview, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(RGID, <span class="dt">level =</span> <span class="kw">c</span>(<span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>)), <span class="dt">y =</span> reads_post_trim_total, <span class="dt">color =</span> original_reseq, <span class="dt">shape =</span> original_reseq)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb19-5" title="5"><span class="st">  </span></a>
<a class="sourceLine" id="cb19-6" title="6"><span class="st">   </span><span class="kw">scale_color_viridis_d</span>(<span class="dt">end =</span> <span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb19-7" title="7"></a>
<a class="sourceLine" id="cb19-8" title="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">2</span>, <span class="dt">alpha =</span> <span class="fl">0.6</span>, <span class="kw">aes</span>(<span class="dt">group =</span> ID), <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="fl">0.5</span>)) <span class="op">+</span><span class="st"> </span><span class="co">#make points, transparent, and group samples</span></a>
<a class="sourceLine" id="cb19-9" title="9"><span class="st">  </span></a>
<a class="sourceLine" id="cb19-10" title="10"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">alpha =</span> <span class="fl">0.4</span>, <span class="dt">outlier.shape =</span> <span class="ot">NA</span>) <span class="op">+</span><span class="st"> </span><span class="co">#add boxplots but do not show them in legend</span></a>
<a class="sourceLine" id="cb19-11" title="11"><span class="st">  </span></a>
<a class="sourceLine" id="cb19-12" title="12"><span class="st">  </span><span class="co">#geom_violin(alpha = 0.5) + #or add violin plots</span></a>
<a class="sourceLine" id="cb19-13" title="13"><span class="st">  </span></a>
<a class="sourceLine" id="cb19-14" title="14"><span class="st">  </span></a>
<a class="sourceLine" id="cb19-15" title="15"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">group =</span> ID), <span class="dt">color =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="fl">0.5</span>), <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span><span class="co">#add line to connect samples present in both sequencing runs</span></a>
<a class="sourceLine" id="cb19-16" title="16"><span class="st">  </span></a>
<a class="sourceLine" id="cb19-17" title="17"><span class="st">  </span><span class="co">#stat_summary(fun = median, geom = &quot;point&quot;, size = 5, alpha = 0.3, position = position_dodge(0.5)) +</span></a>
<a class="sourceLine" id="cb19-18" title="18"><span class="st">  </span></a>
<a class="sourceLine" id="cb19-19" title="19"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;sequencing run&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;reads after quality filtering&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#adjust axis labels</span></a>
<a class="sourceLine" id="cb19-20" title="20"><span class="st">  </span></a>
<a class="sourceLine" id="cb19-21" title="21"><span class="st"> </span></a>
<a class="sourceLine" id="cb19-22" title="22"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="co">#remove legend title</span></a>
<a class="sourceLine" id="cb19-23" title="23"><span class="st">  </span></a>
<a class="sourceLine" id="cb19-24" title="24"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>())  <span class="co">#remove x-axis grid lines</span></a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20compare%20amount%20of%20reads%20per%20sequencing%20strategy-1.png" width="672" /></p>
<p>Maybe more informative about the success of the approaches is to compare % reads that mapped.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_bw</span>()) <span class="co">#set theme to black and white</span></a>
<a class="sourceLine" id="cb20-2" title="2"></a>
<a class="sourceLine" id="cb20-3" title="3"><span class="co">#compare mapped reads (y-axis) between sequencing (x-axis, sorted to have original first) and color according to sequencing run</span></a>
<a class="sourceLine" id="cb20-4" title="4"><span class="kw">ggplot</span>(data_overview, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(RGID, <span class="dt">level =</span> <span class="kw">c</span>(<span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>)), <span class="dt">y =</span> reads_mapped_and_paired, <span class="dt">color =</span> original_reseq, <span class="dt">shape =</span> original_reseq)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-5" title="5"><span class="st">  </span></a>
<a class="sourceLine" id="cb20-6" title="6"><span class="st">  </span><span class="kw">scale_color_viridis_d</span>(<span class="dt">end =</span> <span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb20-7" title="7"></a>
<a class="sourceLine" id="cb20-8" title="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">2</span>, <span class="dt">alpha =</span> <span class="fl">0.6</span>, <span class="kw">aes</span>(<span class="dt">group =</span> ID), <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="fl">0.5</span>)) <span class="op">+</span><span class="st"> </span><span class="co">#make points, transparent, and group samples</span></a>
<a class="sourceLine" id="cb20-9" title="9"><span class="st">  </span></a>
<a class="sourceLine" id="cb20-10" title="10"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">alpha =</span> <span class="fl">0.4</span>, <span class="dt">outlier.shape =</span> <span class="ot">NA</span>) <span class="op">+</span><span class="st"> </span><span class="co">#add boxplots but do not show them in legend</span></a>
<a class="sourceLine" id="cb20-11" title="11"><span class="st">  </span></a>
<a class="sourceLine" id="cb20-12" title="12"><span class="st">  </span><span class="co">#geom_violin(alpha = 0.5) + #or add violin plots</span></a>
<a class="sourceLine" id="cb20-13" title="13"><span class="st">  </span></a>
<a class="sourceLine" id="cb20-14" title="14"><span class="st">  </span></a>
<a class="sourceLine" id="cb20-15" title="15"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">group =</span> ID), <span class="dt">color =</span> <span class="st">&quot;grey&quot;</span>, <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="fl">0.5</span>), <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span><span class="co">#add line to connect samples present in both sequencing approaches</span></a>
<a class="sourceLine" id="cb20-16" title="16"><span class="st">  </span></a>
<a class="sourceLine" id="cb20-17" title="17"><span class="st">  </span><span class="co">#stat_summary(fun = median, geom = &quot;point&quot;, size = 5, alpha = 0.3, position = position_dodge(0.5)) +</span></a>
<a class="sourceLine" id="cb20-18" title="18"><span class="st">  </span></a>
<a class="sourceLine" id="cb20-19" title="19"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;sequencing run&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;mapped reads&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#adjust axis labels</span></a>
<a class="sourceLine" id="cb20-20" title="20"><span class="st">  </span></a>
<a class="sourceLine" id="cb20-21" title="21"><span class="st"> </span></a>
<a class="sourceLine" id="cb20-22" title="22"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="co">#remove legend title</span></a>
<a class="sourceLine" id="cb20-23" title="23"><span class="st">  </span></a>
<a class="sourceLine" id="cb20-24" title="24"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>())  <span class="co">#remove x-axis grid lines</span></a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20mapped%20reads%20per%20sequencing%20strategy-1.png" width="672" /></p>
<p>Finally, let’s have a look at the coverage</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" title="1"></a>
<a class="sourceLine" id="cb21-2" title="2"><span class="kw">theme_set</span>(<span class="kw">theme_bw</span>()) <span class="co">#set theme to black and white</span></a>
<a class="sourceLine" id="cb21-3" title="3"></a>
<a class="sourceLine" id="cb21-4" title="4"><span class="co">#relate number of reads (y-axis) to final coverage (x-axis), color according to sequencing run (original vs. resequenced) and sample type, shape according to original run or resequencing</span></a>
<a class="sourceLine" id="cb21-5" title="5"><span class="kw">ggplot</span>(data_overview, <span class="kw">aes</span>(<span class="dt">x =</span> average_coverage, <span class="dt">y =</span> reads_post_trim_total, <span class="dt">color =</span> original_reseq, <span class="dt">shape =</span> original_reseq)) <span class="op">+</span></a>
<a class="sourceLine" id="cb21-6" title="6"><span class="st">  </span></a>
<a class="sourceLine" id="cb21-7" title="7"><span class="st">  </span><span class="kw">scale_color_viridis_d</span>(<span class="dt">end =</span> <span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb21-8" title="8"><span class="st">  </span></a>
<a class="sourceLine" id="cb21-9" title="9"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">group =</span> ID), <span class="dt">color =</span> <span class="st">&quot;lightgrey&quot;</span>, <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span><span class="co">#connect same samples</span></a>
<a class="sourceLine" id="cb21-10" title="10"><span class="st">  </span></a>
<a class="sourceLine" id="cb21-11" title="11"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="kw">aes</span>(<span class="dt">size =</span> reads_mapped_and_paired, <span class="dt">group =</span> ID)) <span class="op">+</span><span class="st"> </span><span class="co">#add points with slight transparency</span></a>
<a class="sourceLine" id="cb21-12" title="12"><span class="st">  </span></a>
<a class="sourceLine" id="cb21-13" title="13"><span class="st">  </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> <span class="dv">1</span>, <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#include vertical line to show coverage cutoff</span></a>
<a class="sourceLine" id="cb21-14" title="14"><span class="st">  </span></a>
<a class="sourceLine" id="cb21-15" title="15"><span class="st">    </span><span class="co">#scale_y_log10(breaks = scales::trans_breaks(&quot;log10&quot;, function(x) 10^x), labels = scales::trans_format(&quot;log10&quot;, scales::math_format(10^.x))) + #make y-axis logarithmic</span></a>
<a class="sourceLine" id="cb21-16" title="16"><span class="st">  </span></a>
<a class="sourceLine" id="cb21-17" title="17"><span class="st">  </span><span class="co">#scale_x_log10() + #make x-axis logarithmic</span></a>
<a class="sourceLine" id="cb21-18" title="18"><span class="st">  </span></a>
<a class="sourceLine" id="cb21-19" title="19"><span class="st">  </span><span class="co">#annotation_logticks() +</span></a>
<a class="sourceLine" id="cb21-20" title="20"><span class="st">  </span></a>
<a class="sourceLine" id="cb21-21" title="21"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;coverage&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;reads (after quality filtering)&quot;</span>, </a>
<a class="sourceLine" id="cb21-22" title="22">       <span class="dt">size =</span> <span class="st">&quot;% reads mapped&quot;</span>, <span class="dt">color =</span><span class="st">&quot;&quot;</span>, <span class="dt">shape =</span> <span class="st">&quot;&quot;</span>) </a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20number%20of%20reads%20and%20final%20coverage-1.png" width="672" /></p>
<p>C-Curve</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_bw</span>()) <span class="co">#set theme to black and white</span></a>
<a class="sourceLine" id="cb22-2" title="2"></a>
<a class="sourceLine" id="cb22-3" title="3"><span class="co">#relate number of distinct reads (y-axis) to total reads (x-axis), color and shape according to original run or resequencing</span></a>
<a class="sourceLine" id="cb22-4" title="4"><span class="kw">ggplot</span>(data_overview, <span class="kw">aes</span>(<span class="dt">x =</span> unpaired_reads_examined_for_deduplication, <span class="dt">y =</span> sequenced_library_complexity, <span class="dt">color =</span> original_reseq, <span class="dt">shape =</span> original_reseq)) <span class="op">+</span></a>
<a class="sourceLine" id="cb22-5" title="5"><span class="st">  </span></a>
<a class="sourceLine" id="cb22-6" title="6"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">group =</span> ID), <span class="dt">color =</span> <span class="st">&quot;lightgrey&quot;</span>, <span class="dt">show.legend =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span><span class="co">#connect same samples</span></a>
<a class="sourceLine" id="cb22-7" title="7"><span class="st">  </span></a>
<a class="sourceLine" id="cb22-8" title="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="kw">aes</span>(<span class="dt">size =</span> average_coverage, <span class="dt">group =</span> ID)) <span class="op">+</span><span class="st"> </span><span class="co">#add points with slight transparency</span></a>
<a class="sourceLine" id="cb22-9" title="9"><span class="st">  </span></a>
<a class="sourceLine" id="cb22-10" title="10"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.minor.x =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="co">#remove x/y-axis grid lines</span></a>
<a class="sourceLine" id="cb22-11" title="11"><span class="st">  </span></a>
<a class="sourceLine" id="cb22-12" title="12"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;total unpaired reads&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;sequenced library complexity&quot;</span>, </a>
<a class="sourceLine" id="cb22-13" title="13">       <span class="dt">size =</span> <span class="st">&quot;coverage&quot;</span>, <span class="dt">color =</span><span class="st">&quot;&quot;</span>, <span class="dt">shape =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#adjust axis and legend labels</span></a>
<a class="sourceLine" id="cb22-14" title="14"><span class="st">  </span></a>
<a class="sourceLine" id="cb22-15" title="15"><span class="st">   </span><span class="kw">scale_color_viridis_d</span>(<span class="dt">end =</span> <span class="fl">0.5</span>)</a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20c-curve-1.png" width="672" /></p>
<p>comparison to total surviving reads from Trimmomatic to distinct mapped reads</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_bw</span>()) <span class="co">#set theme to black and white</span></a>
<a class="sourceLine" id="cb23-2" title="2"></a>
<a class="sourceLine" id="cb23-3" title="3"><span class="co">#relate number of distinct reads (y-axis) to total surviving reads (x-axis), color  and sample type, shape according to original run or resequencing</span></a>
<a class="sourceLine" id="cb23-4" title="4"><span class="kw">ggplot</span>(data_overview, <span class="kw">aes</span>(<span class="dt">x =</span>  reads_post_trim_total, <span class="dt">y =</span> sequenced_library_complexity, <span class="dt">color =</span> original_reseq, <span class="dt">shape =</span> original_reseq)) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-5" title="5"><span class="st">  </span></a>
<a class="sourceLine" id="cb23-6" title="6"><span class="st">  </span><span class="kw">scale_color_viridis_d</span>(<span class="dt">end =</span> <span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb23-7" title="7"><span class="st">  </span></a>
<a class="sourceLine" id="cb23-8" title="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="kw">aes</span>(<span class="dt">size =</span> average_coverage, <span class="dt">group =</span> ID)) <span class="op">+</span><span class="st"> </span><span class="co">#add points with slight transparency</span></a>
<a class="sourceLine" id="cb23-9" title="9"><span class="st">  </span></a>
<a class="sourceLine" id="cb23-10" title="10"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">panel.grid.minor.x =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span><span class="co">#remove x/y-axis grid lines</span></a>
<a class="sourceLine" id="cb23-11" title="11"><span class="st">  </span></a>
<a class="sourceLine" id="cb23-12" title="12"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;total surviving reads (Trimmomatic)&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;sequenced library complexity&quot;</span>, </a>
<a class="sourceLine" id="cb23-13" title="13">       <span class="dt">size =</span> <span class="st">&quot;coverage&quot;</span>, <span class="dt">color =</span><span class="st">&quot;&quot;</span>, <span class="dt">shape =</span> <span class="st">&quot;&quot;</span>)  <span class="co">#adjust axis and legend labels</span></a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20total%20surving%20reads%20to%20distinct%20mapped%20reads-1.png" width="672" /></p>
<p>plot coverage and st.dv.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_bw</span>()) <span class="co">#set theme to black and white</span></a>
<a class="sourceLine" id="cb24-2" title="2"></a>
<a class="sourceLine" id="cb24-3" title="3"><span class="co">#plot coverage (y-axis) including st.dev for every sample (x-axis), color according to original run or resequencing</span></a>
<a class="sourceLine" id="cb24-4" title="4"><span class="kw">ggplot</span>(<span class="kw">filter</span>(data_overview, original_reseq <span class="op">==</span><span class="st"> &quot;original&quot;</span>), <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(ID, average_coverage), <span class="dt">y =</span> average_coverage, <span class="dt">color =</span> original_reseq, <span class="dt">shape =</span> original_reseq)) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-5" title="5"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-6" title="6"><span class="st">  </span><span class="kw">scale_color_viridis_d</span>(<span class="dt">end =</span> <span class="fl">0.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-7" title="7"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-8" title="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">y=</span>average_coverage), <span class="dt">alpha =</span> <span class="fl">0.8</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-9" title="9"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-10" title="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">data =</span> <span class="kw">filter</span>(data_overview, original_reseq <span class="op">==</span><span class="st"> &quot;reseq&quot;</span>), <span class="kw">aes</span>(<span class="dt">y=</span>average_coverage), <span class="dt">alpha =</span> <span class="fl">0.8</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-11" title="11"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-12" title="12"><span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="dt">data =</span> <span class="kw">filter</span>(data_overview, original_reseq <span class="op">==</span><span class="st"> &quot;reseq&quot;</span>), <span class="kw">aes</span>(<span class="dt">ymin =</span> average_coverage <span class="op">-</span><span class="st"> </span>average_coverarge_stdev, <span class="dt">ymax =</span> average_coverage <span class="op">+</span><span class="st"> </span>average_coverarge_stdev), <span class="dt">alpha =</span> <span class="fl">0.8</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-13" title="13"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-14" title="14"><span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymin =</span> average_coverage <span class="op">-</span><span class="st"> </span>average_coverarge_stdev, <span class="dt">ymax =</span> average_coverage <span class="op">+</span><span class="st"> </span>average_coverarge_stdev), <span class="dt">alpha =</span> <span class="fl">0.8</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-15" title="15"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-16" title="16"><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">1</span>, <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#include horizontal line to show coverage cutoff at 1X</span></a>
<a class="sourceLine" id="cb24-17" title="17"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-18" title="18"><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> median_final_coverage, <span class="dt">linetype =</span> <span class="st">&quot;dotted&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-19" title="19"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-20" title="20"><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">10</span>)) <span class="op">+</span><span class="st"> </span><span class="co">#adjust y axis</span></a>
<a class="sourceLine" id="cb24-21" title="21"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-22" title="22"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">8</span>, <span class="dv">10</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb24-23" title="23"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-24" title="24"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;sample&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;coverage (mean+-sd)&quot;</span>, </a>
<a class="sourceLine" id="cb24-25" title="25">       <span class="dt">color =</span><span class="st">&quot;&quot;</span>, <span class="dt">shape =</span> <span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#adjust axis and legend labels</span></a>
<a class="sourceLine" id="cb24-26" title="26"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-27" title="27"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">50</span>, <span class="dt">vjust =</span> <span class="dv">1</span>, <span class="dt">hjust =</span> <span class="dv">1</span>, <span class="dt">size =</span> <span class="dv">8</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb24-28" title="28"><span class="st">  </span></a>
<a class="sourceLine" id="cb24-29" title="29"><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">guide =</span> <span class="kw">guide_axis</span>(<span class="dt">n.dodge=</span><span class="dv">2</span>))  <span class="co">#avoid overlap of x-axis labels</span></a></code></pre></div>
<p><img src="2_data_files/figure-html/mt%20coverage-1.png" width="672" /></p>
</div>
<div id="relatedness-data" class="section level3">
<h3><span class="header-section-number">4.3.2</span> Relatedness data</h3>
<div id="ngsrelate" class="section level4">
<h4><span class="header-section-number">4.3.2.1</span> ngsRelate</h4>
<p>This is an assessment of the ngsRelate results produced by the Nextflow pipleine created by Ben Hume/SeqAna. The data is saved in the folder <code>pipeline</code> in <code>ngsrelate</code> as <em>relatedness.isec.0002.exMito.thinned.ngsrelate.w.sample.names.results</em>. We now use the results excluding replicated samples and the Z chromosome, which are stored in <code>pipeline/no_repeats_no_Z_relatedness_results/nextflow_output/ngsrelate</code>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" title="1">ngsRelate_results &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;vignettes/data/pipeline/no_repeats_no_Z_relatedness_results/nextflow_output/ngsrelate/relatedness.isec.0002.exMito.thinned.ngsrelate.results&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>,  <span class="dt">as.is =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb25-2" title="2"><span class="kw">str</span>(ngsRelate_results)</a>
<a class="sourceLine" id="cb25-3" title="3"><span class="co">#&gt; &#39;data.frame&#39;:    4560 obs. of  35 variables:</span></a>
<a class="sourceLine" id="cb25-4" title="4"><span class="co">#&gt;  $ a                     : int  0 0 0 0 0 0 0 0 0 0 ...</span></a>
<a class="sourceLine" id="cb25-5" title="5"><span class="co">#&gt;  $ b                     : int  1 2 3 4 5 6 7 8 9 10 ...</span></a>
<a class="sourceLine" id="cb25-6" title="6"><span class="co">#&gt;  $ ida                   : chr  &quot;RGID1_S83&quot; &quot;RGID1_S83&quot; &quot;RGID1_S83&quot; &quot;RGID1_S83&quot; ...</span></a>
<a class="sourceLine" id="cb25-7" title="7"><span class="co">#&gt;  $ idb                   : chr  &quot;RGID1_S82&quot; &quot;RGID1_S29&quot; &quot;RGID1_S2&quot; &quot;RGID2_S13&quot; ...</span></a>
<a class="sourceLine" id="cb25-8" title="8"><span class="co">#&gt;  $ nSites                : int  614924 527474 430254 612597 599536 515211 460041 591284 452116 488952 ...</span></a>
<a class="sourceLine" id="cb25-9" title="9"><span class="co">#&gt;  $ J9                    : num  0.823 0.636 0.427 0.863 0.864 ...</span></a>
<a class="sourceLine" id="cb25-10" title="10"><span class="co">#&gt;  $ J8                    : num  0.0 1.4e-05 1.0e-06 0.0 1.0e-06 0.0 0.0 1.0e-06 3.0e-06 0.0 ...</span></a>
<a class="sourceLine" id="cb25-11" title="11"><span class="co">#&gt;  $ J7                    : num  0.0895 0.0603 0.1184 0.0536 0.0468 ...</span></a>
<a class="sourceLine" id="cb25-12" title="12"><span class="co">#&gt;  $ J6                    : num  0 0.208 0.412 0 0 ...</span></a>
<a class="sourceLine" id="cb25-13" title="13"><span class="co">#&gt;  $ J5                    : num  4e-06 0e+00 1e-05 0e+00 0e+00 0e+00 0e+00 0e+00 0e+00 0e+00 ...</span></a>
<a class="sourceLine" id="cb25-14" title="14"><span class="co">#&gt;  $ J4                    : num  0.05385 0 0 0.030893 0.000062 ...</span></a>
<a class="sourceLine" id="cb25-15" title="15"><span class="co">#&gt;  $ J3                    : num  0e+00 2e-06 0e+00 0e+00 0e+00 0e+00 0e+00 2e-06 0e+00 0e+00 ...</span></a>
<a class="sourceLine" id="cb25-16" title="16"><span class="co">#&gt;  $ J2                    : num  0.0336 0.0959 0.0429 0.0528 0.0888 ...</span></a>
<a class="sourceLine" id="cb25-17" title="17"><span class="co">#&gt;  $ J1                    : num  1.0e-06 0.0 4.3e-05 1.0e-06 0.0 0.0 0.0 1.0e-06 0.0 0.0 ...</span></a>
<a class="sourceLine" id="cb25-18" title="18"><span class="co">#&gt;  $ rab                   : num  0.0895 0.0604 0.1185 0.0536 0.0468 ...</span></a>
<a class="sourceLine" id="cb25-19" title="19"><span class="co">#&gt;  $ Fa                    : num  0.0875 0.096 0.0429 0.0837 0.0889 ...</span></a>
<a class="sourceLine" id="cb25-20" title="20"><span class="co">#&gt;  $ Fb                    : num  0.0336 0.3037 0.4548 0.0528 0.0888 ...</span></a>
<a class="sourceLine" id="cb25-21" title="21"><span class="co">#&gt;  $ theta                 : num  0.0448 0.0302 0.0593 0.0268 0.0234 ...</span></a>
<a class="sourceLine" id="cb25-22" title="22"><span class="co">#&gt;  $ inbred_relatedness_1_2: num  1.0e-06 1.0e-06 4.3e-05 1.0e-06 0.0 0.0 0.0 2.0e-06 0.0 0.0 ...</span></a>
<a class="sourceLine" id="cb25-23" title="23"><span class="co">#&gt;  $ inbred_relatedness_2_1: num  3.0e-06 0.0 4.8e-05 1.0e-06 0.0 0.0 0.0 1.0e-06 0.0 0.0 ...</span></a>
<a class="sourceLine" id="cb25-24" title="24"><span class="co">#&gt;  $ fraternity            : num  0.123 0.156 0.161 0.106 0.136 ...</span></a>
<a class="sourceLine" id="cb25-25" title="25"><span class="co">#&gt;  $ identity              : num  1.0e-06 0.0 4.3e-05 1.0e-06 0.0 0.0 0.0 1.0e-06 0.0 0.0 ...</span></a>
<a class="sourceLine" id="cb25-26" title="26"><span class="co">#&gt;  $ zygosity              : num  0.123 0.156 0.161 0.106 0.136 ...</span></a>
<a class="sourceLine" id="cb25-27" title="27"><span class="co">#&gt;  $ X2of3_IDB             : num  0.15 0.26 0.367 0.122 0.136 ...</span></a>
<a class="sourceLine" id="cb25-28" title="28"><span class="co">#&gt;  $ F_diff_a_b            : num  0.026925 -0.103875 -0.205954 0.015446 0.000031 ...</span></a>
<a class="sourceLine" id="cb25-29" title="29"><span class="co">#&gt;  $ loglh                 : num  -969430 -844356 -677157 -969593 -955122 ...</span></a>
<a class="sourceLine" id="cb25-30" title="30"><span class="co">#&gt;  $ nIter                 : int  54 62 78 62 62 54 38 46 42 58 ...</span></a>
<a class="sourceLine" id="cb25-31" title="31"><span class="co">#&gt;  $ bestoptimll           : int  -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 ...</span></a>
<a class="sourceLine" id="cb25-32" title="32"><span class="co">#&gt;  $ coverage              : num  0.926 0.794 0.648 0.922 0.903 ...</span></a>
<a class="sourceLine" id="cb25-33" title="33"><span class="co">#&gt;  $ X2dsfs                : chr  &quot;3.136819e-01,1.279287e-01,4.005387e-02,1.011532e-01,1.812662e-01,4.215345e-02,4.391939e-02,6.001794e-02,8.982536e-02&quot; &quot;3.378250e-01,7.470870e-02,6.303665e-02,1.394846e-01,1.300943e-01,6.827356e-02,5.710377e-02,3.065345e-02,9.882001e-02&quot; &quot;3.561888e-01,5.169090e-02,6.016002e-02,1.556270e-01,1.122636e-01,8.533396e-02,5.128981e-02,1.941323e-02,1.080326e-01&quot; &quot;3.059626e-01,1.306980e-01,4.481334e-02,1.026846e-01,1.786007e-01,4.361379e-02,4.825899e-02,5.768462e-02,8.768329e-02&quot; ...</span></a>
<a class="sourceLine" id="cb25-34" title="34"><span class="co">#&gt;  $ R0                    : num  0.463 0.923 0.993 0.521 0.581 ...</span></a>
<a class="sourceLine" id="cb25-35" title="35"><span class="co">#&gt;  $ R1                    : num  0.437 0.3 0.265 0.418 0.401 ...</span></a>
<a class="sourceLine" id="cb25-36" title="36"><span class="co">#&gt;  $ KING                  : num  0.0192 -0.1922 -0.2062 -0.0109 -0.0414 ...</span></a>
<a class="sourceLine" id="cb25-37" title="37"><span class="co">#&gt;  $ X2dsfs_loglike        : num  -1205546 -1035930 -823225 -1210883 -1188795 ...</span></a>
<a class="sourceLine" id="cb25-38" title="38"><span class="co">#&gt;  $ X2dsfsf_niter         : int  10 10 10 9 10 10 10 10 10 10 ...</span></a></code></pre></div>
<p>Relatedness was estimated on an average of 4.675586810^{5} sites and a mean coverage of 0.7040709. On average, relatedness is 0.0580915 with a range of 2.3810^{-4}, 0.262088.</p>
<p>We join together the relatedness estimates with the sample information, to know the identity of the dyads</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" title="1"><span class="co">#rename &quot;ida&quot; and &quot;idb&quot; to &quot;RGSM_a&quot; and &quot;RGSM-b&quot;</span></a>
<a class="sourceLine" id="cb26-2" title="2">ngsRelate_results &lt;-<span class="st"> </span><span class="kw">rename</span>(ngsRelate_results, <span class="st">&quot;RGSM_a&quot;</span> =<span class="st"> &quot;ida&quot;</span>)</a>
<a class="sourceLine" id="cb26-3" title="3">ngsRelate_results &lt;-<span class="st"> </span><span class="kw">rename</span>(ngsRelate_results, <span class="st">&quot;RGSM_b&quot;</span> =<span class="st"> &quot;idb&quot;</span>)</a>
<a class="sourceLine" id="cb26-4" title="4"></a>
<a class="sourceLine" id="cb26-5" title="5"><span class="co">#create a new column that includes RGSM_dyad</span></a>
<a class="sourceLine" id="cb26-6" title="6">ngsRelate_results<span class="op">$</span>RGSM_dyad &lt;-<span class="st"> </span><span class="kw">if_else</span>(ngsRelate_results<span class="op">$</span>RGSM_a <span class="op">&lt;</span><span class="st"> </span>ngsRelate_results<span class="op">$</span>RGSM_b, <span class="kw">paste</span>(ngsRelate_results<span class="op">$</span>RGSM_a, ngsRelate_results<span class="op">$</span>RGSM_b, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>), <span class="kw">paste</span>(ngsRelate_results<span class="op">$</span>RGSM_b, ngsRelate_results<span class="op">$</span>RGSM_a, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>))</a>
<a class="sourceLine" id="cb26-7" title="7"></a>
<a class="sourceLine" id="cb26-8" title="8"><span class="co">#first include ID of individual a</span></a>
<a class="sourceLine" id="cb26-9" title="9">ngsRelate_results_ID &lt;-<span class="st"> </span><span class="kw">left_join</span>(ngsRelate_results, seqlib_overview[<span class="dv">6</span><span class="op">:</span><span class="dv">7</span>], <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;RGSM_a&quot;</span> =<span class="st"> &quot;RGSM&quot;</span>))</a>
<a class="sourceLine" id="cb26-10" title="10"></a>
<a class="sourceLine" id="cb26-11" title="11"><span class="co">#rename &quot;ID&quot; to &quot;ID_a&quot;</span></a>
<a class="sourceLine" id="cb26-12" title="12">ngsRelate_results_ID &lt;-<span class="st"> </span><span class="kw">rename</span>(ngsRelate_results_ID, <span class="st">&quot;ID_a&quot;</span> =<span class="st"> &quot;ID&quot;</span>)</a>
<a class="sourceLine" id="cb26-13" title="13"></a>
<a class="sourceLine" id="cb26-14" title="14"><span class="co">#now include ID of individual b</span></a>
<a class="sourceLine" id="cb26-15" title="15">ngsRelate_results_ID &lt;-<span class="st"> </span><span class="kw">left_join</span>(ngsRelate_results_ID, seqlib_overview[<span class="dv">6</span><span class="op">:</span><span class="dv">7</span>], <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;RGSM_b&quot;</span> =<span class="st"> &quot;RGSM&quot;</span>))</a>
<a class="sourceLine" id="cb26-16" title="16"></a>
<a class="sourceLine" id="cb26-17" title="17"><span class="co">#rename &quot;ID&quot; to &quot;ID_b&quot;</span></a>
<a class="sourceLine" id="cb26-18" title="18">ngsRelate_results_ID &lt;-<span class="st"> </span><span class="kw">rename</span>(ngsRelate_results_ID, <span class="st">&quot;ID_b&quot;</span> =<span class="st"> &quot;ID&quot;</span>)</a>
<a class="sourceLine" id="cb26-19" title="19"></a>
<a class="sourceLine" id="cb26-20" title="20"><span class="co">#create a new column that includes the dyad</span></a>
<a class="sourceLine" id="cb26-21" title="21">ngsRelate_results_ID<span class="op">$</span>dyad &lt;-<span class="st"> </span><span class="kw">if_else</span>(ngsRelate_results_ID<span class="op">$</span>ID_a <span class="op">&lt;</span><span class="st"> </span>ngsRelate_results_ID<span class="op">$</span>ID_b, <span class="kw">paste</span>(ngsRelate_results_ID<span class="op">$</span>ID_a, ngsRelate_results_ID<span class="op">$</span>ID_b, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>), <span class="kw">paste</span>(ngsRelate_results_ID<span class="op">$</span>ID_b, ngsRelate_results_ID<span class="op">$</span>ID_a, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>))</a>
<a class="sourceLine" id="cb26-22" title="22"></a>
<a class="sourceLine" id="cb26-23" title="23"><span class="co">#remove duplicates ##maybe not necessary anymore</span></a>
<a class="sourceLine" id="cb26-24" title="24"></a>
<a class="sourceLine" id="cb26-25" title="25">ngsRelate_results_ID &lt;-<span class="st"> </span><span class="kw">distinct</span>(ngsRelate_results_ID)</a></code></pre></div>
<p>We also include the group information</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" title="1"><span class="co">#first include group of individual a</span></a>
<a class="sourceLine" id="cb27-2" title="2">ngsRelate_results_sample_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(ngsRelate_results_ID, sample_info, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;ID_a&quot;</span> =<span class="st"> &quot;ID&quot;</span>))</a>
<a class="sourceLine" id="cb27-3" title="3"></a>
<a class="sourceLine" id="cb27-4" title="4"><span class="co">#rename &quot;Group_ID&quot; to &quot;a_Group&quot; and &quot;sex&quot; to &quot;a_sex&quot;</span></a>
<a class="sourceLine" id="cb27-5" title="5">ngsRelate_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(ngsRelate_results_sample_info, <span class="st">&quot;a_Group&quot;</span> =<span class="st"> &quot;Group_ID&quot;</span>)</a>
<a class="sourceLine" id="cb27-6" title="6">ngsRelate_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(ngsRelate_results_sample_info, <span class="st">&quot;a_sex&quot;</span> =<span class="st"> &quot;sex&quot;</span>)</a>
<a class="sourceLine" id="cb27-7" title="7"></a>
<a class="sourceLine" id="cb27-8" title="8"><span class="co">#then include group of individual b</span></a>
<a class="sourceLine" id="cb27-9" title="9">ngsRelate_results_sample_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(ngsRelate_results_sample_info, sample_info, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;ID_b&quot;</span> =<span class="st"> &quot;ID&quot;</span>))</a>
<a class="sourceLine" id="cb27-10" title="10"></a>
<a class="sourceLine" id="cb27-11" title="11"><span class="co">#rename &quot;Group_ID&quot; to &quot;b_Group&quot; and &quot;sex&quot; to &quot;b_sex&quot;</span></a>
<a class="sourceLine" id="cb27-12" title="12">ngsRelate_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(ngsRelate_results_sample_info, <span class="st">&quot;b_Group&quot;</span> =<span class="st"> &quot;Group_ID&quot;</span>)</a>
<a class="sourceLine" id="cb27-13" title="13">ngsRelate_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(ngsRelate_results_sample_info, <span class="st">&quot;b_sex&quot;</span> =<span class="st"> &quot;sex&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" title="1"><span class="co">#which group IDs are there?</span></a>
<a class="sourceLine" id="cb28-2" title="2"><span class="kw">sort</span>(<span class="kw">unique</span>(ngsRelate_results_sample_info<span class="op">$</span>a_Group))</a>
<a class="sourceLine" id="cb28-3" title="3"><span class="co">#&gt;  [1] &quot;5509&quot;         &quot;5512&quot;         &quot;5931-5938&quot;    &quot;dump&quot;         &quot;mpala&quot;       </span></a>
<a class="sourceLine" id="cb28-4" title="4"><span class="co">#&gt;  [6] &quot;mpala_chicks&quot; &quot;ROOP&quot;         &quot;RRWB&quot;         &quot;wing_tags&quot;    &quot;wt025&quot;</span></a></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" title="1"><span class="kw">sort</span>(<span class="kw">unique</span>(ngsRelate_results_sample_info<span class="op">$</span>b_Group))</a>
<a class="sourceLine" id="cb29-2" title="2"><span class="co">#&gt;  [1] &quot;5509&quot;         &quot;5512&quot;         &quot;5931-5938&quot;    &quot;dump&quot;         &quot;mpala&quot;       </span></a>
<a class="sourceLine" id="cb29-3" title="3"><span class="co">#&gt;  [6] &quot;mpala_chicks&quot; &quot;ROOP&quot;         &quot;RRWB&quot;         &quot;wing_tags&quot;    &quot;wt025&quot;</span></a></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" title="1"><span class="co">#create new column that specifies if individuals are from same or different groups</span></a>
<a class="sourceLine" id="cb30-2" title="2">ngsRelate_results_sample_info &lt;-<span class="st"> </span>ngsRelate_results_sample_info <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb30-3" title="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">dyad_group =</span> <span class="kw">case_when</span>(ID_a <span class="op">==</span><span class="st"> </span>ID_b <span class="op">~</span><span class="st"> &quot;identical&quot;</span>,</a>
<a class="sourceLine" id="cb30-4" title="4">                                a_Group <span class="op">==</span><span class="st"> </span>b_Group <span class="op">~</span><span class="st"> &quot;intragroup&quot;</span>,</a>
<a class="sourceLine" id="cb30-5" title="5">                                a_Group <span class="op">!=</span><span class="st"> </span>b_Group <span class="op">~</span><span class="st"> &quot;intergroup&quot;</span>))</a>
<a class="sourceLine" id="cb30-6" title="6"></a>
<a class="sourceLine" id="cb30-7" title="7"><span class="kw">unique</span>(ngsRelate_results_sample_info<span class="op">$</span>dyad_group)</a>
<a class="sourceLine" id="cb30-8" title="8"><span class="co">#&gt; [1] &quot;intragroup&quot; &quot;intergroup&quot;</span></a></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" title="1"><span class="co">#create new column that specifies if individuals have same or different sex</span></a>
<a class="sourceLine" id="cb31-2" title="2">ngsRelate_results_sample_info &lt;-<span class="st"> </span>ngsRelate_results_sample_info <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb31-3" title="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">dyad_sex =</span> <span class="kw">case_when</span>(ID_a <span class="op">==</span><span class="st"> </span>ID_b <span class="op">~</span><span class="st"> &quot;identical&quot;</span>,</a>
<a class="sourceLine" id="cb31-4" title="4">                                a_sex <span class="op">==</span><span class="st"> </span>b_sex <span class="op">~</span><span class="st"> &quot;same sex&quot;</span>,</a>
<a class="sourceLine" id="cb31-5" title="5">                                a_sex <span class="op">!=</span><span class="st"> </span>b_sex <span class="op">~</span><span class="st"> &quot;male-female&quot;</span>)) </a>
<a class="sourceLine" id="cb31-6" title="6"></a>
<a class="sourceLine" id="cb31-7" title="7"><span class="co">#recode dyads for individuals with undetermined sex to NA</span></a>
<a class="sourceLine" id="cb31-8" title="8">ngsRelate_results_sample_info<span class="op">$</span>dyad_sex[ngsRelate_results_sample_info<span class="op">$</span>a_sex <span class="op">==</span><span class="st"> &quot;u&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb31-9" title="9">ngsRelate_results_sample_info<span class="op">$</span>dyad_sex[ngsRelate_results_sample_info<span class="op">$</span>b_sex <span class="op">==</span><span class="st"> &quot;u&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb31-10" title="10"></a>
<a class="sourceLine" id="cb31-11" title="11"><span class="co">#recode same sex dyads to male or female</span></a>
<a class="sourceLine" id="cb31-12" title="12">ngsRelate_results_sample_info<span class="op">$</span>dyad_sex[ngsRelate_results_sample_info<span class="op">$</span>dyad_sex <span class="op">==</span><span class="st"> &quot;same sex&quot;</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb31-13" title="13"><span class="st">                                         </span>ngsRelate_results_sample_info<span class="op">$</span>a_sex <span class="op">==</span><span class="st"> &quot;f&quot;</span>] &lt;-<span class="st"> &quot;female&quot;</span></a>
<a class="sourceLine" id="cb31-14" title="14">ngsRelate_results_sample_info<span class="op">$</span>dyad_sex[ngsRelate_results_sample_info<span class="op">$</span>dyad_sex <span class="op">==</span><span class="st"> &quot;same sex&quot;</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb31-15" title="15"><span class="st">                                         </span>ngsRelate_results_sample_info<span class="op">$</span>a_sex <span class="op">==</span><span class="st"> &quot;m&quot;</span>] &lt;-<span class="st"> &quot;male&quot;</span></a>
<a class="sourceLine" id="cb31-16" title="16"></a>
<a class="sourceLine" id="cb31-17" title="17"></a>
<a class="sourceLine" id="cb31-18" title="18"><span class="kw">unique</span>(ngsRelate_results_sample_info<span class="op">$</span>dyad_sex)</a>
<a class="sourceLine" id="cb31-19" title="19"><span class="co">#&gt; [1] &quot;male-female&quot; NA            &quot;female&quot;      &quot;male&quot;</span></a></code></pre></div>
<p>Let’s have a closer look at the samples that were sequenced twice</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" title="1"><span class="co">#find samples that were sequenced twice</span></a>
<a class="sourceLine" id="cb32-2" title="2">duplicates &lt;-<span class="st"> </span><span class="kw">filter</span>(ngsRelate_results_sample_info, ID_a <span class="op">==</span><span class="st"> </span>ID_b)</a>
<a class="sourceLine" id="cb32-3" title="3"></a>
<a class="sourceLine" id="cb32-4" title="4"><span class="co">#average relatedness of these duplicted inidividuals</span></a>
<a class="sourceLine" id="cb32-5" title="5"><span class="kw">mean</span>(duplicates<span class="op">$</span>rab)</a>
<a class="sourceLine" id="cb32-6" title="6"><span class="co">#&gt; [1] NaN</span></a>
<a class="sourceLine" id="cb32-7" title="7"></a>
<a class="sourceLine" id="cb32-8" title="8"><span class="co">#range</span></a>
<a class="sourceLine" id="cb32-9" title="9"><span class="kw">range</span>(duplicates<span class="op">$</span>rab)</a>
<a class="sourceLine" id="cb32-10" title="10"><span class="co">#&gt; Warning in min(x): kein nicht-fehlendes Argument für min; gebe Inf zurück</span></a>
<a class="sourceLine" id="cb32-11" title="11"><span class="co">#&gt; Warning in max(x): kein nicht-fehlendes Argument für max; gebe -Inf zurück</span></a>
<a class="sourceLine" id="cb32-12" title="12"><span class="co">#&gt; [1]  Inf -Inf</span></a></code></pre></div>
<p>Create subset with at least 250k SNPs</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" title="1">ngsRelate_results_sample_info_250k &lt;-<span class="st"> </span><span class="kw">filter</span>(ngsRelate_results_sample_info, nSites <span class="op">&gt;=</span><span class="st"> </span><span class="dv">250000</span>)</a>
<a class="sourceLine" id="cb33-2" title="2"></a>
<a class="sourceLine" id="cb33-3" title="3"><span class="kw">unique</span>(ngsRelate_results_sample_info_250k<span class="op">$</span>dyad_group)</a>
<a class="sourceLine" id="cb33-4" title="4"><span class="co">#&gt; [1] &quot;intragroup&quot; &quot;intergroup&quot;</span></a></code></pre></div>
<p>To get an overview of the distribution of relatedness coefficients, we plot pairwise relatedness for all dyads</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" title="1"></a>
<a class="sourceLine" id="cb34-2" title="2"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb34-3" title="3"></a>
<a class="sourceLine" id="cb34-4" title="4"><span class="co">#plot relatedness rab (y-axis) for every dyad (x-axis), ordered from low to high relatedness and colored according to inter- or intragroup dyad</span></a>
<a class="sourceLine" id="cb34-5" title="5"></a>
<a class="sourceLine" id="cb34-6" title="6"><span class="kw">ggplot</span>(ngsRelate_results_sample_info_250k, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(dyad, rab), <span class="dt">y =</span> rab, <span class="dt">color =</span> dyad_sex, <span class="dt">shape =</span> dyad_group)) <span class="op">+</span></a>
<a class="sourceLine" id="cb34-7" title="7"><span class="st">  </span></a>
<a class="sourceLine" id="cb34-8" title="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>, <span class="kw">aes</span>(<span class="dt">y =</span> rab)) <span class="op">+</span><span class="st"> </span><span class="co">#plot relatedness estimates as points </span></a>
<a class="sourceLine" id="cb34-9" title="9"><span class="st">  </span></a>
<a class="sourceLine" id="cb34-10" title="10"><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="kw">c</span>(<span class="fl">0.125</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span>), <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#include horizontal line to show traditional kinship categories</span></a>
<a class="sourceLine" id="cb34-11" title="11"><span class="st">  </span></a>
<a class="sourceLine" id="cb34-12" title="12"><span class="st">  </span><span class="kw">labs</span>( <span class="dt">x =</span> <span class="st">&quot;Dyad&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;Relatedness (ngsRelate)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb34-13" title="13"><span class="st">  </span></a>
<a class="sourceLine" id="cb34-14" title="14"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;distribution of relatedness (ngsrelate) 250k&quot;</span>)</a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20dyadic%20relatedness%20(ngsrelate)%20distribution-1.png" width="672" /></p>
<p>check correlation between number of SNPs and sex</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb35-2" title="2"></a>
<a class="sourceLine" id="cb35-3" title="3"><span class="co">#plot nSites (y-axis) and for the different categories of sex of dyad (x-axis) colored according to coverage</span></a>
<a class="sourceLine" id="cb35-4" title="4"></a>
<a class="sourceLine" id="cb35-5" title="5"><span class="kw">ggplot</span>(ngsRelate_results_sample_info, <span class="kw">aes</span>(<span class="dt">x =</span> dyad_sex, <span class="dt">y =</span> nSites, <span class="dt">color =</span> coverage, <span class="dt">shape =</span> dyad_group)) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-6" title="6"><span class="st">  </span></a>
<a class="sourceLine" id="cb35-7" title="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-8" title="8"><span class="st">  </span></a>
<a class="sourceLine" id="cb35-9" title="9"><span class="st">  </span><span class="kw">geom_boxplot</span>()</a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20(ngsrelate)%20dyad_sex%20and%20nSites-1.png" width="672" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" title="1"></a>
<a class="sourceLine" id="cb36-2" title="2"> <span class="kw">ggtitle</span>(<span class="st">&quot;nsites vs sex&quot;</span>)</a>
<a class="sourceLine" id="cb36-3" title="3"><span class="co">#&gt; $title</span></a>
<a class="sourceLine" id="cb36-4" title="4"><span class="co">#&gt; [1] &quot;nsites vs sex&quot;</span></a>
<a class="sourceLine" id="cb36-5" title="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb36-6" title="6"><span class="co">#&gt; attr(,&quot;class&quot;)</span></a>
<a class="sourceLine" id="cb36-7" title="7"><span class="co">#&gt; [1] &quot;labels&quot;</span></a></code></pre></div>
<p>Compare relatedness estimates according to group and sex</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb37-2" title="2"></a>
<a class="sourceLine" id="cb37-3" title="3"><span class="co">#plot relatedness rab (y-axis) for the different categories of sex of dyad (x-axis) colored according to group membership</span></a>
<a class="sourceLine" id="cb37-4" title="4"></a>
<a class="sourceLine" id="cb37-5" title="5"><span class="kw">ggplot</span>(ngsRelate_results_sample_info_250k, <span class="kw">aes</span>(<span class="dt">x =</span> dyad_sex, <span class="dt">y =</span> rab, <span class="dt">color =</span> dyad_group, <span class="dt">shape =</span> dyad_group)) <span class="op">+</span></a>
<a class="sourceLine" id="cb37-6" title="6"><span class="st">  </span></a>
<a class="sourceLine" id="cb37-7" title="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>, <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="fl">0.5</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb37-8" title="8"><span class="st">  </span></a>
<a class="sourceLine" id="cb37-9" title="9"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">alpha =</span> <span class="fl">0.4</span>, <span class="dt">outlier.shape =</span> <span class="ot">NA</span>)</a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20(ngsrelate)%20relatedness%20for%20group%20and%20sex-1.png" width="672" /></p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" title="1"></a>
<a class="sourceLine" id="cb38-2" title="2"> <span class="kw">ggtitle</span>(<span class="st">&quot;relatedness (ngsrelate) 250k&quot;</span>)</a>
<a class="sourceLine" id="cb38-3" title="3"><span class="co">#&gt; $title</span></a>
<a class="sourceLine" id="cb38-4" title="4"><span class="co">#&gt; [1] &quot;relatedness (ngsrelate) 250k&quot;</span></a>
<a class="sourceLine" id="cb38-5" title="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb38-6" title="6"><span class="co">#&gt; attr(,&quot;class&quot;)</span></a>
<a class="sourceLine" id="cb38-7" title="7"><span class="co">#&gt; [1] &quot;labels&quot;</span></a></code></pre></div>
</div>
<div id="lcmlkin" class="section level4">
<h4><span class="header-section-number">4.3.2.2</span> lcmlkin</h4>
<p>We now use the results excluding replicated samples, which are stored in <code>pipeline/no_repeats_relatedness_results/nf_results/lcmlkin</code></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" title="1">lcmlkin_results &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;vignettes/data/pipeline/no_repeats_relatedness_results/nf_results/lcmlkin/relatedness.isec.0002.exMito.thinned.lcmlkin.results&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>,  <span class="dt">as.is =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb39-2" title="2"><span class="kw">str</span>(lcmlkin_results)</a>
<a class="sourceLine" id="cb39-3" title="3"><span class="co">#&gt; &#39;data.frame&#39;:    4560 obs. of  7 variables:</span></a>
<a class="sourceLine" id="cb39-4" title="4"><span class="co">#&gt;  $ Ind1  : chr  &quot;RGID1_S83&quot; &quot;RGID1_S83&quot; &quot;RGID1_S83&quot; &quot;RGID1_S83&quot; ...</span></a>
<a class="sourceLine" id="cb39-5" title="5"><span class="co">#&gt;  $ Ind2  : chr  &quot;RGID1_S49&quot; &quot;RGID2_S13&quot; &quot;RGID1_S89&quot; &quot;RGID2_S11&quot; ...</span></a>
<a class="sourceLine" id="cb39-6" title="6"><span class="co">#&gt;  $ k0_hat: num  0.911 0.878 0.835 0.815 0.863 0.904 0.814 0.816 0.877 0.903 ...</span></a>
<a class="sourceLine" id="cb39-7" title="7"><span class="co">#&gt;  $ k1_hat: num  0 0.002 0.004 0.003 0.002 0.003 0.005 0.003 0.002 0.003 ...</span></a>
<a class="sourceLine" id="cb39-8" title="8"><span class="co">#&gt;  $ k2_hat: num  0.089 0.12 0.162 0.182 0.134 0.093 0.182 0.18 0.121 0.094 ...</span></a>
<a class="sourceLine" id="cb39-9" title="9"><span class="co">#&gt;  $ pi_HAT: num  0.089 0.121 0.164 0.183 0.136 0.094 0.184 0.182 0.122 0.096 ...</span></a>
<a class="sourceLine" id="cb39-10" title="10"><span class="co">#&gt;  $ nbSNP : int  617110 729018 615902 622500 713042 579737 518643 623265 704472 603659 ...</span></a></code></pre></div>
<p>We join together the relatedness estimates with the sample information, to know the identity of the dyads</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" title="1"><span class="co">#rename &quot;ida&quot; and &quot;idb&quot; to &quot;RGSM_a&quot; and &quot;RGSM-b&quot;</span></a>
<a class="sourceLine" id="cb40-2" title="2">lcmlkin_results &lt;-<span class="st"> </span><span class="kw">rename</span>(lcmlkin_results, <span class="st">&quot;RGSM_a&quot;</span> =<span class="st"> &quot;Ind1&quot;</span>)</a>
<a class="sourceLine" id="cb40-3" title="3">lcmlkin_results &lt;-<span class="st"> </span><span class="kw">rename</span>(lcmlkin_results, <span class="st">&quot;RGSM_b&quot;</span> =<span class="st"> &quot;Ind2&quot;</span>)</a>
<a class="sourceLine" id="cb40-4" title="4"></a>
<a class="sourceLine" id="cb40-5" title="5"><span class="co">#create a new column that includes RGSM_dyad</span></a>
<a class="sourceLine" id="cb40-6" title="6">lcmlkin_results<span class="op">$</span>RGSM_dyad &lt;-<span class="st"> </span><span class="kw">if_else</span>(lcmlkin_results<span class="op">$</span>RGSM_a <span class="op">&lt;</span><span class="st"> </span>lcmlkin_results<span class="op">$</span>RGSM_b, <span class="kw">paste</span>(lcmlkin_results<span class="op">$</span>RGSM_a, lcmlkin_results<span class="op">$</span>RGSM_b, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>), <span class="kw">paste</span>(lcmlkin_results<span class="op">$</span>RGSM_b, lcmlkin_results<span class="op">$</span>RGSM_a, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>))</a>
<a class="sourceLine" id="cb40-7" title="7"></a>
<a class="sourceLine" id="cb40-8" title="8"><span class="co">#first include ID of individual a</span></a>
<a class="sourceLine" id="cb40-9" title="9">lcmlkin_results_ID &lt;-<span class="st"> </span><span class="kw">left_join</span>(lcmlkin_results, seqlib_overview[<span class="dv">6</span><span class="op">:</span><span class="dv">7</span>], <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;RGSM_a&quot;</span> =<span class="st"> &quot;RGSM&quot;</span>))</a>
<a class="sourceLine" id="cb40-10" title="10"></a>
<a class="sourceLine" id="cb40-11" title="11"><span class="co">#rename &quot;ID&quot; to &quot;ID_a&quot;</span></a>
<a class="sourceLine" id="cb40-12" title="12">lcmlkin_results_ID &lt;-<span class="st"> </span><span class="kw">rename</span>(lcmlkin_results_ID, <span class="st">&quot;ID_a&quot;</span> =<span class="st"> &quot;ID&quot;</span>)</a>
<a class="sourceLine" id="cb40-13" title="13"></a>
<a class="sourceLine" id="cb40-14" title="14"><span class="co">#now include ID of individual b</span></a>
<a class="sourceLine" id="cb40-15" title="15">lcmlkin_results_ID &lt;-<span class="st"> </span><span class="kw">left_join</span>(lcmlkin_results_ID, seqlib_overview[<span class="dv">6</span><span class="op">:</span><span class="dv">7</span>], <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;RGSM_b&quot;</span> =<span class="st"> &quot;RGSM&quot;</span>))</a>
<a class="sourceLine" id="cb40-16" title="16"></a>
<a class="sourceLine" id="cb40-17" title="17"><span class="co">#rename &quot;ID&quot; to &quot;ID_b&quot;</span></a>
<a class="sourceLine" id="cb40-18" title="18">lcmlkin_results_ID &lt;-<span class="st"> </span><span class="kw">rename</span>(lcmlkin_results_ID, <span class="st">&quot;ID_b&quot;</span> =<span class="st"> &quot;ID&quot;</span>)</a>
<a class="sourceLine" id="cb40-19" title="19"></a>
<a class="sourceLine" id="cb40-20" title="20"><span class="co">#create a new column that includes the dyad</span></a>
<a class="sourceLine" id="cb40-21" title="21">lcmlkin_results_ID<span class="op">$</span>dyad &lt;-<span class="st"> </span><span class="kw">if_else</span>(lcmlkin_results_ID<span class="op">$</span>ID_a <span class="op">&lt;</span><span class="st"> </span>lcmlkin_results_ID<span class="op">$</span>ID_b, <span class="kw">paste</span>(lcmlkin_results_ID<span class="op">$</span>ID_a, lcmlkin_results_ID<span class="op">$</span>ID_b, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>), <span class="kw">paste</span>(lcmlkin_results_ID<span class="op">$</span>ID_b, lcmlkin_results_ID<span class="op">$</span>ID_a, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>))</a>
<a class="sourceLine" id="cb40-22" title="22"></a>
<a class="sourceLine" id="cb40-23" title="23"><span class="co">#remove duplicates ##maybe not necessary anymore</span></a>
<a class="sourceLine" id="cb40-24" title="24"></a>
<a class="sourceLine" id="cb40-25" title="25">lcmlkin_results_ID &lt;-<span class="st"> </span><span class="kw">distinct</span>(lcmlkin_results_ID)</a></code></pre></div>
<p>We also include the group information</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" title="1"><span class="co">#first include group of individual a</span></a>
<a class="sourceLine" id="cb41-2" title="2">lcmlkin_results_sample_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(lcmlkin_results_ID, sample_info, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;ID_a&quot;</span> =<span class="st"> &quot;ID&quot;</span>))</a>
<a class="sourceLine" id="cb41-3" title="3"></a>
<a class="sourceLine" id="cb41-4" title="4"><span class="co">#rename &quot;Group_ID&quot; to &quot;a_Group&quot;</span></a>
<a class="sourceLine" id="cb41-5" title="5">lcmlkin_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(lcmlkin_results_sample_info, <span class="st">&quot;a_Group&quot;</span> =<span class="st"> &quot;Group_ID&quot;</span>)</a>
<a class="sourceLine" id="cb41-6" title="6">lcmlkin_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(lcmlkin_results_sample_info, <span class="st">&quot;a_sex&quot;</span> =<span class="st"> &quot;sex&quot;</span>)</a>
<a class="sourceLine" id="cb41-7" title="7"></a>
<a class="sourceLine" id="cb41-8" title="8"><span class="co">#then include group of individual b</span></a>
<a class="sourceLine" id="cb41-9" title="9">lcmlkin_results_sample_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(lcmlkin_results_sample_info, sample_info, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;ID_b&quot;</span> =<span class="st"> &quot;ID&quot;</span>))</a>
<a class="sourceLine" id="cb41-10" title="10">lcmlkin_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(lcmlkin_results_sample_info, <span class="st">&quot;b_sex&quot;</span> =<span class="st"> &quot;sex&quot;</span>)</a>
<a class="sourceLine" id="cb41-11" title="11"></a>
<a class="sourceLine" id="cb41-12" title="12"><span class="co">#rename &quot;Group_ID&quot; to &quot;b_Group&quot;</span></a>
<a class="sourceLine" id="cb41-13" title="13">lcmlkin_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(lcmlkin_results_sample_info, <span class="st">&quot;b_Group&quot;</span> =<span class="st"> &quot;Group_ID&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" title="1"><span class="co">#create new column that specifies if individuals are from same or different groups</span></a>
<a class="sourceLine" id="cb42-2" title="2">lcmlkin_results_sample_info &lt;-<span class="st"> </span>lcmlkin_results_sample_info <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb42-3" title="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">dyad_group =</span> <span class="kw">if_else</span>(a_Group <span class="op">==</span><span class="st"> </span>b_Group, <span class="st">&quot;intra&quot;</span>, <span class="st">&quot;inter&quot;</span>))</a>
<a class="sourceLine" id="cb42-4" title="4"><span class="kw">str</span>(lcmlkin_results_sample_info)</a>
<a class="sourceLine" id="cb42-5" title="5"><span class="co">#&gt; &#39;data.frame&#39;:    4560 obs. of  16 variables:</span></a>
<a class="sourceLine" id="cb42-6" title="6"><span class="co">#&gt;  $ RGSM_a    : chr  &quot;RGID1_S83&quot; &quot;RGID1_S83&quot; &quot;RGID1_S83&quot; &quot;RGID1_S83&quot; ...</span></a>
<a class="sourceLine" id="cb42-7" title="7"><span class="co">#&gt;  $ RGSM_b    : chr  &quot;RGID1_S49&quot; &quot;RGID2_S13&quot; &quot;RGID1_S89&quot; &quot;RGID2_S11&quot; ...</span></a>
<a class="sourceLine" id="cb42-8" title="8"><span class="co">#&gt;  $ k0_hat    : num  0.911 0.878 0.835 0.815 0.863 0.904 0.814 0.816 0.877 0.903 ...</span></a>
<a class="sourceLine" id="cb42-9" title="9"><span class="co">#&gt;  $ k1_hat    : num  0 0.002 0.004 0.003 0.002 0.003 0.005 0.003 0.002 0.003 ...</span></a>
<a class="sourceLine" id="cb42-10" title="10"><span class="co">#&gt;  $ k2_hat    : num  0.089 0.12 0.162 0.182 0.134 0.093 0.182 0.18 0.121 0.094 ...</span></a>
<a class="sourceLine" id="cb42-11" title="11"><span class="co">#&gt;  $ pi_HAT    : num  0.089 0.121 0.164 0.183 0.136 0.094 0.184 0.182 0.122 0.096 ...</span></a>
<a class="sourceLine" id="cb42-12" title="12"><span class="co">#&gt;  $ nbSNP     : int  617110 729018 615902 622500 713042 579737 518643 623265 704472 603659 ...</span></a>
<a class="sourceLine" id="cb42-13" title="13"><span class="co">#&gt;  $ RGSM_dyad : chr  &quot;RGID1_S49-RGID1_S83&quot; &quot;RGID1_S83-RGID2_S13&quot; &quot;RGID1_S83-RGID1_S89&quot; &quot;RGID1_S83-RGID2_S11&quot; ...</span></a>
<a class="sourceLine" id="cb42-14" title="14"><span class="co">#&gt;  $ ID_a      : chr  &quot;WT00386&quot; &quot;WT00386&quot; &quot;WT00386&quot; &quot;WT00386&quot; ...</span></a>
<a class="sourceLine" id="cb42-15" title="15"><span class="co">#&gt;  $ ID_b      : chr  &quot;A1307&quot; &quot;A1312&quot; &quot;W1671&quot; &quot;W1673&quot; ...</span></a>
<a class="sourceLine" id="cb42-16" title="16"><span class="co">#&gt;  $ dyad      : chr  &quot;A1307-WT00386&quot; &quot;A1312-WT00386&quot; &quot;W1671-WT00386&quot; &quot;W1673-WT00386&quot; ...</span></a>
<a class="sourceLine" id="cb42-17" title="17"><span class="co">#&gt;  $ a_Group   : chr  &quot;mpala_chicks&quot; &quot;mpala_chicks&quot; &quot;mpala_chicks&quot; &quot;mpala_chicks&quot; ...</span></a>
<a class="sourceLine" id="cb42-18" title="18"><span class="co">#&gt;  $ a_sex     : chr  &quot;f&quot; &quot;f&quot; &quot;f&quot; &quot;f&quot; ...</span></a>
<a class="sourceLine" id="cb42-19" title="19"><span class="co">#&gt;  $ b_Group   : chr  &quot;wing_tags&quot; &quot;wing_tags&quot; &quot;wt025&quot; &quot;wt025&quot; ...</span></a>
<a class="sourceLine" id="cb42-20" title="20"><span class="co">#&gt;  $ b_sex     : chr  &quot;m&quot; &quot;m&quot; &quot;f&quot; &quot;f&quot; ...</span></a>
<a class="sourceLine" id="cb42-21" title="21"><span class="co">#&gt;  $ dyad_group: chr  &quot;inter&quot; &quot;inter&quot; &quot;inter&quot; &quot;inter&quot; ...</span></a></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" title="1"><span class="co">#create new column that specifies if individuals have same or different sex</span></a>
<a class="sourceLine" id="cb43-2" title="2">lcmlkin_results_sample_info &lt;-<span class="st"> </span>lcmlkin_results_sample_info <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb43-3" title="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">dyad_sex =</span> <span class="kw">case_when</span>(ID_a <span class="op">==</span><span class="st"> </span>ID_b <span class="op">~</span><span class="st"> &quot;identical&quot;</span>,</a>
<a class="sourceLine" id="cb43-4" title="4">                                a_sex <span class="op">==</span><span class="st"> </span>b_sex <span class="op">~</span><span class="st"> &quot;same sex&quot;</span>,</a>
<a class="sourceLine" id="cb43-5" title="5">                                a_sex <span class="op">!=</span><span class="st"> </span>b_sex <span class="op">~</span><span class="st"> &quot;male-female&quot;</span>)) </a>
<a class="sourceLine" id="cb43-6" title="6"></a>
<a class="sourceLine" id="cb43-7" title="7"><span class="co">#recode dyads for individuals with undetermined sex to NA</span></a>
<a class="sourceLine" id="cb43-8" title="8">lcmlkin_results_sample_info<span class="op">$</span>dyad_sex[lcmlkin_results_sample_info<span class="op">$</span>a_sex <span class="op">==</span><span class="st"> &quot;u&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb43-9" title="9">lcmlkin_results_sample_info<span class="op">$</span>dyad_sex[lcmlkin_results_sample_info<span class="op">$</span>b_sex <span class="op">==</span><span class="st"> &quot;u&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb43-10" title="10"></a>
<a class="sourceLine" id="cb43-11" title="11"><span class="co">#recode same sex dyads to male or female</span></a>
<a class="sourceLine" id="cb43-12" title="12">lcmlkin_results_sample_info<span class="op">$</span>dyad_sex[lcmlkin_results_sample_info<span class="op">$</span>dyad_sex <span class="op">==</span><span class="st"> &quot;same sex&quot;</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-13" title="13"><span class="st">                                         </span>lcmlkin_results_sample_info<span class="op">$</span>a_sex <span class="op">==</span><span class="st"> &quot;f&quot;</span>] &lt;-<span class="st"> &quot;female&quot;</span></a>
<a class="sourceLine" id="cb43-14" title="14">lcmlkin_results_sample_info<span class="op">$</span>dyad_sex[lcmlkin_results_sample_info<span class="op">$</span>dyad_sex <span class="op">==</span><span class="st"> &quot;same sex&quot;</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb43-15" title="15"><span class="st">                                         </span>lcmlkin_results_sample_info<span class="op">$</span>a_sex <span class="op">==</span><span class="st"> &quot;m&quot;</span>] &lt;-<span class="st"> &quot;male&quot;</span></a>
<a class="sourceLine" id="cb43-16" title="16"></a>
<a class="sourceLine" id="cb43-17" title="17"></a>
<a class="sourceLine" id="cb43-18" title="18"><span class="kw">unique</span>(lcmlkin_results_sample_info<span class="op">$</span>dyad_sex)</a>
<a class="sourceLine" id="cb43-19" title="19"><span class="co">#&gt; [1] &quot;male-female&quot; &quot;female&quot;      NA            &quot;male&quot;</span></a></code></pre></div>
<p>Let’s have a closer look at the samples that were sequenced twice</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" title="1"><span class="co">#find samples that were sequenced twice</span></a>
<a class="sourceLine" id="cb44-2" title="2">duplicates_lcmlkin &lt;-<span class="st"> </span><span class="kw">filter</span>(lcmlkin_results_sample_info, ID_a <span class="op">==</span><span class="st"> </span>ID_b)</a>
<a class="sourceLine" id="cb44-3" title="3"></a>
<a class="sourceLine" id="cb44-4" title="4"><span class="co">#average relatedness of these duplicted inidividuals</span></a>
<a class="sourceLine" id="cb44-5" title="5"><span class="kw">mean</span>(duplicates_lcmlkin<span class="op">$</span>pi_HAT)</a>
<a class="sourceLine" id="cb44-6" title="6"><span class="co">#&gt; [1] NaN</span></a>
<a class="sourceLine" id="cb44-7" title="7"></a>
<a class="sourceLine" id="cb44-8" title="8"><span class="co">#range</span></a>
<a class="sourceLine" id="cb44-9" title="9"><span class="kw">range</span>(duplicates_lcmlkin<span class="op">$</span>pi_HAT)</a>
<a class="sourceLine" id="cb44-10" title="10"><span class="co">#&gt; Warning in min(x): kein nicht-fehlendes Argument für min; gebe Inf zurück</span></a>
<a class="sourceLine" id="cb44-11" title="11"><span class="co">#&gt; Warning in max(x): kein nicht-fehlendes Argument für max; gebe -Inf zurück</span></a>
<a class="sourceLine" id="cb44-12" title="12"><span class="co">#&gt; [1]  Inf -Inf</span></a></code></pre></div>
<p>Create subset with at least 300k SNPs</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" title="1">lcmlkin_results_sample_info_300k &lt;-<span class="st"> </span><span class="kw">filter</span>(lcmlkin_results_sample_info, nbSNP <span class="op">&gt;=</span><span class="st"> </span><span class="dv">300000</span>)</a></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" title="1"></a>
<a class="sourceLine" id="cb46-2" title="2"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb46-3" title="3"></a>
<a class="sourceLine" id="cb46-4" title="4"><span class="co">#plot relatedness pi_HAT (y-axis) for every dyad (x-axis), ordered from low to high relatedness and colored according to inter- or intragroup dyad</span></a>
<a class="sourceLine" id="cb46-5" title="5"></a>
<a class="sourceLine" id="cb46-6" title="6"><span class="kw">ggplot</span>(lcmlkin_results_sample_info_300k, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(dyad, pi_HAT), <span class="dt">y =</span> pi_HAT, <span class="dt">color =</span> dyad_sex, <span class="dt">shape =</span> dyad_group)) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-7" title="7"><span class="st">  </span></a>
<a class="sourceLine" id="cb46-8" title="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="kw">aes</span>(<span class="dt">y =</span> pi_HAT)) <span class="op">+</span><span class="st"> </span><span class="co">#plot relatedness estimates as points </span></a>
<a class="sourceLine" id="cb46-9" title="9"><span class="st">  </span></a>
<a class="sourceLine" id="cb46-10" title="10"><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="kw">c</span>(<span class="fl">0.125</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span>), <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co">#include horizontal line to show traditional kinship categories</span></a>
<a class="sourceLine" id="cb46-11" title="11"><span class="st">  </span></a>
<a class="sourceLine" id="cb46-12" title="12"><span class="st">  </span><span class="kw">labs</span>( <span class="dt">x =</span> <span class="st">&quot;Dyad&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;Relatedness (lcmlkin)&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb46-13" title="13"></a>
<a class="sourceLine" id="cb46-14" title="14"><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;distribution of relatedness (lcmlkin) 300k SNPs&quot;</span>)</a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20dyadic%20relatedness%20(lcmlkin)%20distribution-1.png" width="672" /></p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" title="1"></a>
<a class="sourceLine" id="cb47-2" title="2"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb47-3" title="3"></a>
<a class="sourceLine" id="cb47-4" title="4"><span class="co">#plot relatedness pi_HAT (y-axis) and corresponding k0 (x-axis) colored according to nbSNP</span></a>
<a class="sourceLine" id="cb47-5" title="5"></a>
<a class="sourceLine" id="cb47-6" title="6"><span class="kw">ggplot</span>(lcmlkin_results_sample_info, <span class="kw">aes</span>(<span class="dt">x =</span> k0_hat, <span class="dt">y =</span> pi_HAT, <span class="dt">color =</span> nbSNP, <span class="dt">shape =</span> dyad_group)) <span class="op">+</span></a>
<a class="sourceLine" id="cb47-7" title="7"><span class="st">  </span></a>
<a class="sourceLine" id="cb47-8" title="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb47-9" title="9"></a>
<a class="sourceLine" id="cb47-10" title="10"><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">&quot;lcmlkin k0 vs phi&quot;</span>)</a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20dyadic%20relatedness%20(lcmlkin)%20distribution%20and%20k0-1.png" width="672" /></p>
<p>check correlation between number of SNPs and sex</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb48-2" title="2"></a>
<a class="sourceLine" id="cb48-3" title="3"><span class="co">#plot nbSNPs (y-axis) and for the different categories of sex of dyad (x-axis) colored according to dyad_group</span></a>
<a class="sourceLine" id="cb48-4" title="4"></a>
<a class="sourceLine" id="cb48-5" title="5"><span class="kw">ggplot</span>(lcmlkin_results_sample_info, <span class="kw">aes</span>(<span class="dt">x =</span> dyad_sex, <span class="dt">y =</span> nbSNP, <span class="dt">color =</span> dyad_group, <span class="dt">shape =</span> dyad_group)) <span class="op">+</span></a>
<a class="sourceLine" id="cb48-6" title="6"><span class="st">  </span></a>
<a class="sourceLine" id="cb48-7" title="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb48-8" title="8"><span class="st">  </span></a>
<a class="sourceLine" id="cb48-9" title="9"><span class="st">  </span><span class="kw">geom_boxplot</span>()</a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20(lcmlkin)%20dyad_sex%20and%20nbSNP-1.png" width="672" /></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" title="1"></a>
<a class="sourceLine" id="cb49-2" title="2"> <span class="kw">ggtitle</span>(<span class="st">&quot;nbSNP vs sex of dyad&quot;</span>)</a>
<a class="sourceLine" id="cb49-3" title="3"><span class="co">#&gt; $title</span></a>
<a class="sourceLine" id="cb49-4" title="4"><span class="co">#&gt; [1] &quot;nbSNP vs sex of dyad&quot;</span></a>
<a class="sourceLine" id="cb49-5" title="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb49-6" title="6"><span class="co">#&gt; attr(,&quot;class&quot;)</span></a>
<a class="sourceLine" id="cb49-7" title="7"><span class="co">#&gt; [1] &quot;labels&quot;</span></a></code></pre></div>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb50-2" title="2"></a>
<a class="sourceLine" id="cb50-3" title="3"><span class="co">#plot relatedness pi_HAT (y-axis) for the different categories of sex of dyad (x-axis) colored according to group membership</span></a>
<a class="sourceLine" id="cb50-4" title="4"></a>
<a class="sourceLine" id="cb50-5" title="5"><span class="kw">ggplot</span>(lcmlkin_results_sample_info_300k, <span class="kw">aes</span>(<span class="dt">x =</span> dyad_group, <span class="dt">y =</span> pi_HAT, <span class="dt">color =</span> dyad_sex)) <span class="op">+</span></a>
<a class="sourceLine" id="cb50-6" title="6"><span class="st">  </span></a>
<a class="sourceLine" id="cb50-7" title="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb50-8" title="8"><span class="st">  </span></a>
<a class="sourceLine" id="cb50-9" title="9"><span class="st">  </span><span class="kw">geom_boxplot</span>()</a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20(lcmlkin%20relatedness%20for%20group%20and%20sex-1.png" width="672" /></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" title="1"></a>
<a class="sourceLine" id="cb51-2" title="2"> <span class="kw">ggtitle</span>(<span class="st">&quot;relatedness (lcmlkin) 300k&quot;</span>)</a>
<a class="sourceLine" id="cb51-3" title="3"><span class="co">#&gt; $title</span></a>
<a class="sourceLine" id="cb51-4" title="4"><span class="co">#&gt; [1] &quot;relatedness (lcmlkin) 300k&quot;</span></a>
<a class="sourceLine" id="cb51-5" title="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb51-6" title="6"><span class="co">#&gt; attr(,&quot;class&quot;)</span></a>
<a class="sourceLine" id="cb51-7" title="7"><span class="co">#&gt; [1] &quot;labels&quot;</span></a></code></pre></div>
</div>
<div id="read" class="section level4">
<h4><span class="header-section-number">4.3.2.3</span> READ</h4>
<p>This is an assessment of the READ results produced by the Nextflow pipleine created by Ben Hume/SeqAna. The data is saved in the folder <code>pipeline</code> in <code>read</code> as <em>meansP0_AncientDNA_normalized</em>. The preprocessed data (categorizes unrelated, first- and second-order related) is sotred as <em>READ_results</em>.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" title="1">READ_rawresults &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;vignettes/data/pipeline/no_repeats_relatedness_results/nf_results/read/meansP0_AncientDNA_normalized&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot; &quot;</span>, <span class="dt">as.is =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb52-2" title="2"><span class="kw">str</span>(READ_rawresults)</a>
<a class="sourceLine" id="cb52-3" title="3"><span class="co">#&gt; &#39;data.frame&#39;:    4560 obs. of  5 variables:</span></a>
<a class="sourceLine" id="cb52-4" title="4"><span class="co">#&gt;  $ PairIndividuals            : chr  &quot;RGID1_S10RGID1_S11&quot; &quot;RGID1_S10RGID1_S26&quot; &quot;RGID1_S10RGID1_S36&quot; &quot;RGID1_S10RGID1_S39&quot; ...</span></a>
<a class="sourceLine" id="cb52-5" title="5"><span class="co">#&gt;  $ Normalized2AlleleDifference: num  0.954 0.968 0.931 0.972 0.985 ...</span></a>
<a class="sourceLine" id="cb52-6" title="6"><span class="co">#&gt;  $ StandardError              : num  0.00477 0.00498 0.00534 0.00472 0.00489 ...</span></a>
<a class="sourceLine" id="cb52-7" title="7"><span class="co">#&gt;  $ NonNormalizedP0            : num  0.341 0.346 0.333 0.347 0.352 ...</span></a>
<a class="sourceLine" id="cb52-8" title="8"><span class="co">#&gt;  $ NonNormalizedStandardError : num  0.0017 0.00178 0.00191 0.00169 0.00175 ...</span></a></code></pre></div>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" title="1">READ_kinresults &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;vignettes/data/pipeline/no_repeats_relatedness_results/nf_results/read/READ_results&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">as.is =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb53-2" title="2"><span class="kw">str</span>(READ_kinresults)</a>
<a class="sourceLine" id="cb53-3" title="3"><span class="co">#&gt; &#39;data.frame&#39;:    4560 obs. of  4 variables:</span></a>
<a class="sourceLine" id="cb53-4" title="4"><span class="co">#&gt;  $ PairIndividuals: chr  &quot;RGID1_S10RGID1_S11&quot; &quot;RGID1_S10RGID1_S26&quot; &quot;RGID1_S10RGID1_S36&quot; &quot;RGID1_S10RGID1_S39&quot; ...</span></a>
<a class="sourceLine" id="cb53-5" title="5"><span class="co">#&gt;  $ Relationship   : chr  &quot;Unrelated&quot; &quot;Unrelated&quot; &quot;Unrelated&quot; &quot;Unrelated&quot; ...</span></a>
<a class="sourceLine" id="cb53-6" title="6"><span class="co">#&gt;  $ Z_upper        : num  NA NA NA NA NA NA NA NA NA NA ...</span></a>
<a class="sourceLine" id="cb53-7" title="7"><span class="co">#&gt;  $ Z_lower        : num  -9.91 -12.35 -4.68 -13.99 -16.15 ...</span></a></code></pre></div>
<p>Merge these two result files</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" title="1">READ_results &lt;-<span class="st"> </span><span class="kw">left_join</span>(READ_rawresults, READ_kinresults, <span class="dt">by =</span> <span class="st">&quot;PairIndividuals&quot;</span>)</a>
<a class="sourceLine" id="cb54-2" title="2"><span class="kw">str</span>(READ_results)</a>
<a class="sourceLine" id="cb54-3" title="3"><span class="co">#&gt; &#39;data.frame&#39;:    4560 obs. of  8 variables:</span></a>
<a class="sourceLine" id="cb54-4" title="4"><span class="co">#&gt;  $ PairIndividuals            : chr  &quot;RGID1_S10RGID1_S11&quot; &quot;RGID1_S10RGID1_S26&quot; &quot;RGID1_S10RGID1_S36&quot; &quot;RGID1_S10RGID1_S39&quot; ...</span></a>
<a class="sourceLine" id="cb54-5" title="5"><span class="co">#&gt;  $ Normalized2AlleleDifference: num  0.954 0.968 0.931 0.972 0.985 ...</span></a>
<a class="sourceLine" id="cb54-6" title="6"><span class="co">#&gt;  $ StandardError              : num  0.00477 0.00498 0.00534 0.00472 0.00489 ...</span></a>
<a class="sourceLine" id="cb54-7" title="7"><span class="co">#&gt;  $ NonNormalizedP0            : num  0.341 0.346 0.333 0.347 0.352 ...</span></a>
<a class="sourceLine" id="cb54-8" title="8"><span class="co">#&gt;  $ NonNormalizedStandardError : num  0.0017 0.00178 0.00191 0.00169 0.00175 ...</span></a>
<a class="sourceLine" id="cb54-9" title="9"><span class="co">#&gt;  $ Relationship               : chr  &quot;Unrelated&quot; &quot;Unrelated&quot; &quot;Unrelated&quot; &quot;Unrelated&quot; ...</span></a>
<a class="sourceLine" id="cb54-10" title="10"><span class="co">#&gt;  $ Z_upper                    : num  NA NA NA NA NA NA NA NA NA NA ...</span></a>
<a class="sourceLine" id="cb54-11" title="11"><span class="co">#&gt;  $ Z_lower                    : num  -9.91 -12.35 -4.68 -13.99 -16.15 ...</span></a></code></pre></div>
<p>Now we need to split the <em>PairIndividuals</em> into individual 1 and 2, identify these individuals and then create the dyads corresponding to the previous results.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" title="1"><span class="co">#Need to split PairIndividuals into 2 new columns</span></a>
<a class="sourceLine" id="cb55-2" title="2">READ_results<span class="op">$</span>RGSM_a &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;(RGID[[:alnum:]]{1}_S[[:alnum:]]{1,2})(RGID[[:alnum:]]{1}_S[[:alnum:]]{1,2})&quot;</span>, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">1&quot;</span>, READ_results<span class="op">$</span>PairIndividuals)</a>
<a class="sourceLine" id="cb55-3" title="3"></a>
<a class="sourceLine" id="cb55-4" title="4">READ_results<span class="op">$</span>RGSM_b &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;(RGID[[:alnum:]]{1}_S[[:alnum:]]{1,2})(RGID[[:alnum:]]{1}_S[[:alnum:]]{1,2})&quot;</span>, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">2&quot;</span>, READ_results<span class="op">$</span>PairIndividuals)</a>
<a class="sourceLine" id="cb55-5" title="5"></a>
<a class="sourceLine" id="cb55-6" title="6"><span class="co">#create a new column that includes RGSM_dyad</span></a>
<a class="sourceLine" id="cb55-7" title="7">READ_results<span class="op">$</span>RGSM_dyad &lt;-<span class="st"> </span><span class="kw">if_else</span>(READ_results<span class="op">$</span>RGSM_a <span class="op">&lt;</span><span class="st"> </span>READ_results<span class="op">$</span>RGSM_b, <span class="kw">paste</span>(READ_results<span class="op">$</span>RGSM_a, READ_results<span class="op">$</span>RGSM_b, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>), <span class="kw">paste</span>(READ_results<span class="op">$</span>RGSM_b, READ_results<span class="op">$</span>RGSM_a, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>))</a>
<a class="sourceLine" id="cb55-8" title="8"></a>
<a class="sourceLine" id="cb55-9" title="9"><span class="kw">str</span>(READ_results)</a>
<a class="sourceLine" id="cb55-10" title="10"><span class="co">#&gt; &#39;data.frame&#39;:    4560 obs. of  11 variables:</span></a>
<a class="sourceLine" id="cb55-11" title="11"><span class="co">#&gt;  $ PairIndividuals            : chr  &quot;RGID1_S10RGID1_S11&quot; &quot;RGID1_S10RGID1_S26&quot; &quot;RGID1_S10RGID1_S36&quot; &quot;RGID1_S10RGID1_S39&quot; ...</span></a>
<a class="sourceLine" id="cb55-12" title="12"><span class="co">#&gt;  $ Normalized2AlleleDifference: num  0.954 0.968 0.931 0.972 0.985 ...</span></a>
<a class="sourceLine" id="cb55-13" title="13"><span class="co">#&gt;  $ StandardError              : num  0.00477 0.00498 0.00534 0.00472 0.00489 ...</span></a>
<a class="sourceLine" id="cb55-14" title="14"><span class="co">#&gt;  $ NonNormalizedP0            : num  0.341 0.346 0.333 0.347 0.352 ...</span></a>
<a class="sourceLine" id="cb55-15" title="15"><span class="co">#&gt;  $ NonNormalizedStandardError : num  0.0017 0.00178 0.00191 0.00169 0.00175 ...</span></a>
<a class="sourceLine" id="cb55-16" title="16"><span class="co">#&gt;  $ Relationship               : chr  &quot;Unrelated&quot; &quot;Unrelated&quot; &quot;Unrelated&quot; &quot;Unrelated&quot; ...</span></a>
<a class="sourceLine" id="cb55-17" title="17"><span class="co">#&gt;  $ Z_upper                    : num  NA NA NA NA NA NA NA NA NA NA ...</span></a>
<a class="sourceLine" id="cb55-18" title="18"><span class="co">#&gt;  $ Z_lower                    : num  -9.91 -12.35 -4.68 -13.99 -16.15 ...</span></a>
<a class="sourceLine" id="cb55-19" title="19"><span class="co">#&gt;  $ RGSM_a                     : chr  &quot;RGID1_S10&quot; &quot;RGID1_S10&quot; &quot;RGID1_S10&quot; &quot;RGID1_S10&quot; ...</span></a>
<a class="sourceLine" id="cb55-20" title="20"><span class="co">#&gt;  $ RGSM_b                     : chr  &quot;RGID1_S11&quot; &quot;RGID1_S26&quot; &quot;RGID1_S36&quot; &quot;RGID1_S39&quot; ...</span></a>
<a class="sourceLine" id="cb55-21" title="21"><span class="co">#&gt;  $ RGSM_dyad                  : chr  &quot;RGID1_S10-RGID1_S11&quot; &quot;RGID1_S10-RGID1_S26&quot; &quot;RGID1_S10-RGID1_S36&quot; &quot;RGID1_S10-RGID1_S39&quot; ...</span></a></code></pre></div>
<p>We join together the relatedness estimates with the sample information, to know the identity of the dyads</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" title="1"></a>
<a class="sourceLine" id="cb56-2" title="2"><span class="co">#first include ID of individual a</span></a>
<a class="sourceLine" id="cb56-3" title="3">READ_results_ID &lt;-<span class="st"> </span><span class="kw">left_join</span>(READ_results, seqlib_overview[<span class="dv">6</span><span class="op">:</span><span class="dv">7</span>], <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;RGSM_a&quot;</span> =<span class="st"> &quot;RGSM&quot;</span>))</a>
<a class="sourceLine" id="cb56-4" title="4"></a>
<a class="sourceLine" id="cb56-5" title="5"><span class="co">#rename &quot;ID&quot; to &quot;ID_a&quot;</span></a>
<a class="sourceLine" id="cb56-6" title="6">READ_results_ID &lt;-<span class="st"> </span><span class="kw">rename</span>(READ_results_ID, <span class="st">&quot;ID_a&quot;</span> =<span class="st"> &quot;ID&quot;</span>)</a>
<a class="sourceLine" id="cb56-7" title="7"></a>
<a class="sourceLine" id="cb56-8" title="8"><span class="co">#now include ID of individual b</span></a>
<a class="sourceLine" id="cb56-9" title="9">READ_results_ID &lt;-<span class="st"> </span><span class="kw">left_join</span>(READ_results_ID, seqlib_overview[<span class="dv">6</span><span class="op">:</span><span class="dv">7</span>], <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;RGSM_b&quot;</span> =<span class="st"> &quot;RGSM&quot;</span>))</a>
<a class="sourceLine" id="cb56-10" title="10"></a>
<a class="sourceLine" id="cb56-11" title="11"><span class="co">#rename &quot;ID&quot; to &quot;ID_b&quot;</span></a>
<a class="sourceLine" id="cb56-12" title="12">READ_results_ID &lt;-<span class="st"> </span><span class="kw">rename</span>(READ_results_ID, <span class="st">&quot;ID_b&quot;</span> =<span class="st"> &quot;ID&quot;</span>)</a>
<a class="sourceLine" id="cb56-13" title="13"></a>
<a class="sourceLine" id="cb56-14" title="14"><span class="co">#create a new column that includes the dyad</span></a>
<a class="sourceLine" id="cb56-15" title="15">READ_results_ID<span class="op">$</span>dyad &lt;-<span class="st"> </span><span class="kw">if_else</span>(READ_results_ID<span class="op">$</span>ID_a <span class="op">&lt;</span><span class="st"> </span>READ_results_ID<span class="op">$</span>ID_b, <span class="kw">paste</span>(READ_results_ID<span class="op">$</span>ID_a, READ_results_ID<span class="op">$</span>ID_b, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>), <span class="kw">paste</span>(READ_results_ID<span class="op">$</span>ID_b, READ_results_ID<span class="op">$</span>ID_a, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>))</a>
<a class="sourceLine" id="cb56-16" title="16"></a>
<a class="sourceLine" id="cb56-17" title="17"><span class="co">#remove duplicates ##maybe not necessary anymore</span></a>
<a class="sourceLine" id="cb56-18" title="18"></a>
<a class="sourceLine" id="cb56-19" title="19">READ_results_ID &lt;-<span class="st"> </span><span class="kw">distinct</span>(READ_results_ID)</a></code></pre></div>
<p>Let’s have a closer look at the samples that were sequenced twice</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" title="1"><span class="co">#find samples that were sequenced twice</span></a>
<a class="sourceLine" id="cb57-2" title="2">duplicates &lt;-<span class="st"> </span><span class="kw">filter</span>(READ_results_ID, ID_a <span class="op">==</span><span class="st"> </span>ID_b)</a>
<a class="sourceLine" id="cb57-3" title="3"></a>
<a class="sourceLine" id="cb57-4" title="4"><span class="co">#average relatedness of these duplicted inidividuals</span></a>
<a class="sourceLine" id="cb57-5" title="5"><span class="kw">mean</span>(duplicates<span class="op">$</span>NonNormalizedP0)</a>
<a class="sourceLine" id="cb57-6" title="6"><span class="co">#&gt; [1] NaN</span></a>
<a class="sourceLine" id="cb57-7" title="7"></a>
<a class="sourceLine" id="cb57-8" title="8"><span class="co">#range</span></a>
<a class="sourceLine" id="cb57-9" title="9"><span class="kw">range</span>(duplicates<span class="op">$</span>NonNormalizedP0)</a>
<a class="sourceLine" id="cb57-10" title="10"><span class="co">#&gt; Warning in min(x): kein nicht-fehlendes Argument für min; gebe Inf zurück</span></a>
<a class="sourceLine" id="cb57-11" title="11"><span class="co">#&gt; Warning in max(x): kein nicht-fehlendes Argument für max; gebe -Inf zurück</span></a>
<a class="sourceLine" id="cb57-12" title="12"><span class="co">#&gt; [1]  Inf -Inf</span></a></code></pre></div>
<p>We also include the group information</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" title="1"><span class="co">#first include group of individual a</span></a>
<a class="sourceLine" id="cb58-2" title="2">READ_results_sample_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(READ_results_ID, sample_info, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;ID_a&quot;</span> =<span class="st"> &quot;ID&quot;</span>))</a>
<a class="sourceLine" id="cb58-3" title="3"></a>
<a class="sourceLine" id="cb58-4" title="4"><span class="co">#rename &quot;Group_ID&quot; to &quot;a_Group&quot;</span></a>
<a class="sourceLine" id="cb58-5" title="5">READ_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(READ_results_sample_info, <span class="st">&quot;a_Group&quot;</span> =<span class="st"> &quot;Group_ID&quot;</span>)</a>
<a class="sourceLine" id="cb58-6" title="6">READ_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(READ_results_sample_info, <span class="st">&quot;a_sex&quot;</span> =<span class="st"> &quot;sex&quot;</span>)</a>
<a class="sourceLine" id="cb58-7" title="7"></a>
<a class="sourceLine" id="cb58-8" title="8"><span class="co">#then include group of individual b</span></a>
<a class="sourceLine" id="cb58-9" title="9">READ_results_sample_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(READ_results_sample_info, sample_info, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;ID_b&quot;</span> =<span class="st"> &quot;ID&quot;</span>))</a>
<a class="sourceLine" id="cb58-10" title="10">READ_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(READ_results_sample_info, <span class="st">&quot;b_sex&quot;</span> =<span class="st"> &quot;sex&quot;</span>)</a>
<a class="sourceLine" id="cb58-11" title="11"></a>
<a class="sourceLine" id="cb58-12" title="12"><span class="co">#rename &quot;Group_ID&quot; to &quot;b_Group&quot;</span></a>
<a class="sourceLine" id="cb58-13" title="13">READ_results_sample_info &lt;-<span class="st"> </span><span class="kw">rename</span>(READ_results_sample_info, <span class="st">&quot;b_Group&quot;</span> =<span class="st"> &quot;Group_ID&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" title="1"><span class="co">#create new column that specifies if individuals are from same or different groups</span></a>
<a class="sourceLine" id="cb59-2" title="2">READ_results_sample_info &lt;-<span class="st"> </span>READ_results_sample_info <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb59-3" title="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">dyad_group =</span> <span class="kw">case_when</span>(ID_a <span class="op">==</span><span class="st"> </span>ID_b <span class="op">~</span><span class="st"> &quot;identical&quot;</span>,</a>
<a class="sourceLine" id="cb59-4" title="4">                                a_Group <span class="op">==</span><span class="st"> </span>b_Group <span class="op">~</span><span class="st"> &quot;intragroup&quot;</span>,</a>
<a class="sourceLine" id="cb59-5" title="5">                                a_Group <span class="op">!=</span><span class="st"> </span>b_Group <span class="op">~</span><span class="st"> &quot;intergroup&quot;</span>))</a>
<a class="sourceLine" id="cb59-6" title="6"></a>
<a class="sourceLine" id="cb59-7" title="7"><span class="kw">unique</span>(READ_results_sample_info<span class="op">$</span>dyad_group)</a>
<a class="sourceLine" id="cb59-8" title="8"><span class="co">#&gt; [1] &quot;intergroup&quot; &quot;intragroup&quot;</span></a></code></pre></div>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" title="1"><span class="co">#create new column that specifies if individuals have same or different sex</span></a>
<a class="sourceLine" id="cb60-2" title="2">READ_results_sample_info &lt;-<span class="st"> </span>READ_results_sample_info <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb60-3" title="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">dyad_sex =</span> <span class="kw">case_when</span>(ID_a <span class="op">==</span><span class="st"> </span>ID_b <span class="op">~</span><span class="st"> &quot;identical&quot;</span>,</a>
<a class="sourceLine" id="cb60-4" title="4">                                a_sex <span class="op">==</span><span class="st"> </span>b_sex <span class="op">~</span><span class="st"> &quot;same sex&quot;</span>,</a>
<a class="sourceLine" id="cb60-5" title="5">                                a_sex <span class="op">!=</span><span class="st"> </span>b_sex <span class="op">~</span><span class="st"> &quot;male-female&quot;</span>)) </a>
<a class="sourceLine" id="cb60-6" title="6"></a>
<a class="sourceLine" id="cb60-7" title="7"><span class="co">#recode dyads for individuals with undetermined sex to NA</span></a>
<a class="sourceLine" id="cb60-8" title="8">READ_results_sample_info<span class="op">$</span>dyad_sex[READ_results_sample_info<span class="op">$</span>a_sex <span class="op">==</span><span class="st"> &quot;u&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb60-9" title="9">READ_results_sample_info<span class="op">$</span>dyad_sex[READ_results_sample_info<span class="op">$</span>b_sex <span class="op">==</span><span class="st"> &quot;u&quot;</span>] &lt;-<span class="st"> </span><span class="ot">NA</span></a>
<a class="sourceLine" id="cb60-10" title="10"></a>
<a class="sourceLine" id="cb60-11" title="11"><span class="co">#recode same sex dyads to male or female</span></a>
<a class="sourceLine" id="cb60-12" title="12">READ_results_sample_info<span class="op">$</span>dyad_sex[READ_results_sample_info<span class="op">$</span>dyad_sex <span class="op">==</span><span class="st"> &quot;same sex&quot;</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-13" title="13"><span class="st">                                         </span>READ_results_sample_info<span class="op">$</span>a_sex <span class="op">==</span><span class="st"> &quot;f&quot;</span>] &lt;-<span class="st"> &quot;female&quot;</span></a>
<a class="sourceLine" id="cb60-14" title="14">READ_results_sample_info<span class="op">$</span>dyad_sex[READ_results_sample_info<span class="op">$</span>dyad_sex <span class="op">==</span><span class="st"> &quot;same sex&quot;</span> <span class="op">&amp;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-15" title="15"><span class="st">                                         </span>READ_results_sample_info<span class="op">$</span>a_sex <span class="op">==</span><span class="st"> &quot;m&quot;</span>] &lt;-<span class="st"> &quot;male&quot;</span></a>
<a class="sourceLine" id="cb60-16" title="16"></a>
<a class="sourceLine" id="cb60-17" title="17"></a>
<a class="sourceLine" id="cb60-18" title="18"><span class="kw">unique</span>(READ_results_sample_info<span class="op">$</span>dyad_sex)</a>
<a class="sourceLine" id="cb60-19" title="19"><span class="co">#&gt; [1] NA            &quot;male-female&quot; &quot;male&quot;        &quot;female&quot;</span></a></code></pre></div>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" title="1"></a>
<a class="sourceLine" id="cb61-2" title="2"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb61-3" title="3"></a>
<a class="sourceLine" id="cb61-4" title="4"><span class="co">#plot relatedness P0 (y-axis) for every dyad (x-axis), ordered from low to high relatedness and colored according to inter- or intragroup dyad</span></a>
<a class="sourceLine" id="cb61-5" title="5"></a>
<a class="sourceLine" id="cb61-6" title="6"><span class="kw">ggplot</span>(READ_results_sample_info, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(PairIndividuals, NonNormalizedP0), <span class="dt">y =</span> NonNormalizedP0, <span class="dt">color =</span> dyad_sex)) <span class="op">+</span></a>
<a class="sourceLine" id="cb61-7" title="7"><span class="st">  </span></a>
<a class="sourceLine" id="cb61-8" title="8"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="kw">aes</span>(<span class="dt">y =</span> NonNormalizedP0)) <span class="op">+</span><span class="st"> </span><span class="co">#plot relatedness estimates as points </span></a>
<a class="sourceLine" id="cb61-9" title="9"><span class="st">  </span></a>
<a class="sourceLine" id="cb61-10" title="10"><span class="st">  </span><span class="kw">labs</span>( <span class="dt">x =</span> <span class="st">&quot;Dyad&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;Relatedness (READ)&quot;</span>) </a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20dyadic%20relatedness%20(READ)%20distribution-1.png" width="672" /></p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" title="1"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb62-2" title="2"></a>
<a class="sourceLine" id="cb62-3" title="3"><span class="co">#plot relatedness pi_HAT (y-axis) for the different categories of sex of dyad (x-axis) colored according to group membership</span></a>
<a class="sourceLine" id="cb62-4" title="4"></a>
<a class="sourceLine" id="cb62-5" title="5"><span class="kw">ggplot</span>(READ_results_sample_info, <span class="kw">aes</span>(<span class="dt">x =</span> dyad_sex, <span class="dt">y =</span> NonNormalizedP0, <span class="dt">color =</span> dyad_group)) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-6" title="6"><span class="st">  </span></a>
<a class="sourceLine" id="cb62-7" title="7"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb62-8" title="8"><span class="st">  </span></a>
<a class="sourceLine" id="cb62-9" title="9"><span class="st">  </span><span class="kw">geom_boxplot</span>()</a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20(READ)%20relatedness%20for%20group%20and%20sex-1.png" width="672" /></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" title="1"></a>
<a class="sourceLine" id="cb63-2" title="2"> <span class="kw">ggtitle</span>(<span class="st">&quot;relatedness (READ)&quot;</span>)</a>
<a class="sourceLine" id="cb63-3" title="3"><span class="co">#&gt; $title</span></a>
<a class="sourceLine" id="cb63-4" title="4"><span class="co">#&gt; [1] &quot;relatedness (READ)&quot;</span></a>
<a class="sourceLine" id="cb63-5" title="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb63-6" title="6"><span class="co">#&gt; attr(,&quot;class&quot;)</span></a>
<a class="sourceLine" id="cb63-7" title="7"><span class="co">#&gt; [1] &quot;labels&quot;</span></a></code></pre></div>
</div>
<div id="combined-relatedness-results" class="section level4">
<h4><span class="header-section-number">4.3.2.4</span> Combined relatedness results</h4>
<p>merge ngsrelate and lcmlkin</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb64-1" title="1">relate_results &lt;-<span class="st"> </span><span class="kw">full_join</span>(ngsRelate_results_sample_info, lcmlkin_results_sample_info, <span class="dt">by =</span> <span class="st">&quot;RGSM_dyad&quot;</span>)</a>
<a class="sourceLine" id="cb64-2" title="2">relate_results &lt;-<span class="st"> </span><span class="kw">full_join</span>(relate_results, READ_results_sample_info, <span class="dt">by =</span> <span class="st">&quot;RGSM_dyad&quot;</span>)</a></code></pre></div>
<p>plot</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb65-1" title="1"></a>
<a class="sourceLine" id="cb65-2" title="2"><span class="kw">theme_set</span>(<span class="kw">theme_classic</span>())</a>
<a class="sourceLine" id="cb65-3" title="3"></a>
<a class="sourceLine" id="cb65-4" title="4"><span class="co">#plot relatedness pi_HAT and rab (y-axis) for every dyad (x-axis), ordered from low to high relatedness and colored according to inter- or intragroup dyad</span></a>
<a class="sourceLine" id="cb65-5" title="5"></a>
<a class="sourceLine" id="cb65-6" title="6"><span class="kw">ggplot</span>(<span class="kw">filter</span>(relate_results, coverage <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.85</span>), <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(dyad, pi_HAT), <span class="dt">color =</span> dyad_sex)) <span class="op">+</span></a>
<a class="sourceLine" id="cb65-7" title="7"><span class="st">  </span></a>
<a class="sourceLine" id="cb65-8" title="8"><span class="st">  </span><span class="kw">scale_colour_viridis_d</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb65-9" title="9"><span class="st">  </span></a>
<a class="sourceLine" id="cb65-10" title="10"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="kw">aes</span>(<span class="dt">y =</span> pi_HAT), <span class="dt">shape =</span> <span class="dv">15</span>) <span class="op">+</span><span class="st"> </span><span class="co">#plot lcmlkin relatedness estimates as points </span></a>
<a class="sourceLine" id="cb65-11" title="11"><span class="st">  </span></a>
<a class="sourceLine" id="cb65-12" title="12"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="kw">aes</span>(<span class="dt">y =</span> <span class="fl">0.7</span><span class="op">-</span>NonNormalizedP0), <span class="dt">shape =</span> <span class="dv">16</span>) <span class="op">+</span><span class="st"> </span><span class="co">#plot READ estimate</span></a>
<a class="sourceLine" id="cb65-13" title="13"><span class="st">  </span></a>
<a class="sourceLine" id="cb65-14" title="14"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>, <span class="kw">aes</span>(<span class="dt">y =</span> rab), <span class="dt">shape =</span> <span class="dv">17</span>) <span class="op">+</span><span class="st"> </span><span class="co"># plot ngsRelate estimates</span></a>
<a class="sourceLine" id="cb65-15" title="15"><span class="st">  </span></a>
<a class="sourceLine" id="cb65-16" title="16"><span class="st">  </span><span class="kw">scale_y_continuous</span>(</a>
<a class="sourceLine" id="cb65-17" title="17">    </a>
<a class="sourceLine" id="cb65-18" title="18">    <span class="dt">name =</span> <span class="st">&quot;relatedness (lcmlkin)&quot;</span>, <span class="co">#features of first axis</span></a>
<a class="sourceLine" id="cb65-19" title="19">    </a>
<a class="sourceLine" id="cb65-20" title="20">    <span class="dt">sec.axis =</span> <span class="kw">sec_axis</span>(<span class="op">~</span>., <span class="dt">name =</span> <span class="st">&quot;P0&quot;</span>) <span class="co">#features of second axis</span></a>
<a class="sourceLine" id="cb65-21" title="21">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb65-22" title="22"><span class="st">  </span></a>
<a class="sourceLine" id="cb65-23" title="23"><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Comparison of relatedness estimators (&gt;0.85X)&quot;</span>)</a>
<a class="sourceLine" id="cb65-24" title="24"><span class="co">#&gt; Warning: Removed 28 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb65-25" title="25"></a>
<a class="sourceLine" id="cb65-26" title="26"><span class="co">#&gt; Warning: Removed 28 rows containing missing values (geom_point).</span></a>
<a class="sourceLine" id="cb65-27" title="27"></a>
<a class="sourceLine" id="cb65-28" title="28"><span class="co">#&gt; Warning: Removed 28 rows containing missing values (geom_point).</span></a></code></pre></div>
<p><img src="2_data_files/figure-html/figure%20dyadic%20relatedness%20distribution-1.png" width="672" /></p>
<p>Compare correlation among the three estimators</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" title="1"><span class="kw">ggplot</span>(relate_results, <span class="kw">aes</span>(<span class="dt">x =</span> rab, <span class="dt">y =</span> pi_HAT, <span class="dt">color =</span> dyad_sex, <span class="dt">size =</span> nSites, <span class="dt">shape =</span> dyad_group.x)) <span class="op">+</span></a>
<a class="sourceLine" id="cb66-2" title="2"><span class="st">  </span></a>
<a class="sourceLine" id="cb66-3" title="3"><span class="st">  </span><span class="kw">scale_colour_viridis_d</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb66-4" title="4"><span class="st">  </span></a>
<a class="sourceLine" id="cb66-5" title="5"><span class="st">   </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>) </a>
<a class="sourceLine" id="cb66-6" title="6"><span class="co">#&gt; Warning: Removed 374 rows containing missing values (geom_point).</span></a></code></pre></div>
<p><img src="2_data_files/figure-html/correlation%20ngsrelate%20lcmlkin-1.png" width="672" /></p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb67-1" title="1"></a>
<a class="sourceLine" id="cb67-2" title="2"><span class="kw">ggplot</span>(relate_results, <span class="kw">aes</span>(<span class="dt">x =</span> rab, <span class="dt">y =</span> NonNormalizedP0, <span class="dt">color =</span> dyad_sex, <span class="dt">size =</span> nSites, <span class="dt">shape =</span> dyad_group.x)) <span class="op">+</span></a>
<a class="sourceLine" id="cb67-3" title="3"><span class="st">  </span></a>
<a class="sourceLine" id="cb67-4" title="4"><span class="st">  </span><span class="kw">scale_colour_viridis_d</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb67-5" title="5"><span class="st">  </span></a>
<a class="sourceLine" id="cb67-6" title="6"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>) </a>
<a class="sourceLine" id="cb67-7" title="7"><span class="co">#&gt; Warning: Removed 374 rows containing missing values (geom_point).</span></a></code></pre></div>
<p><img src="2_data_files/figure-html/correlation%20ngsrelate%20READ-1.png" width="672" /></p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb68-1" title="1"></a>
<a class="sourceLine" id="cb68-2" title="2"><span class="kw">ggplot</span>(relate_results, <span class="kw">aes</span>(<span class="dt">x =</span> pi_HAT, <span class="dt">y =</span> NonNormalizedP0, <span class="dt">color =</span> dyad_sex, <span class="dt">size =</span> nSites, <span class="dt">shape =</span> dyad_group.x)) <span class="op">+</span></a>
<a class="sourceLine" id="cb68-3" title="3"><span class="st">  </span></a>
<a class="sourceLine" id="cb68-4" title="4"><span class="st">  </span><span class="kw">scale_colour_viridis_d</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb68-5" title="5"><span class="st">  </span></a>
<a class="sourceLine" id="cb68-6" title="6"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>) </a>
<a class="sourceLine" id="cb68-7" title="7"><span class="co">#&gt; Warning: Removed 374 rows containing missing values (geom_point).</span></a></code></pre></div>
<p><img src="2_data_files/figure-html/correlation%20lcmlkin%20READ-1.png" width="672" /></p>
<p>Have a closer look at the groups</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb69-1" title="1"><span class="kw">ggplot</span>(<span class="kw">filter</span>(relate_results, dyad_group <span class="op">==</span><span class="st"> &quot;intragroup&quot;</span>), <span class="kw">aes</span>(<span class="dt">x =</span> dyad_sex, <span class="dt">y =</span> rab, <span class="dt">color =</span> dyad_sex)) <span class="op">+</span></a>
<a class="sourceLine" id="cb69-2" title="2"><span class="st">  </span></a>
<a class="sourceLine" id="cb69-3" title="3"><span class="st">  </span><span class="kw">scale_color_viridis_d</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb69-4" title="4"><span class="st">  </span></a>
<a class="sourceLine" id="cb69-5" title="5"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">alpha =</span> <span class="fl">0.2</span>, <span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="fl">0.5</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb69-6" title="6"><span class="st">  </span></a>
<a class="sourceLine" id="cb69-7" title="7"><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">alpha =</span> <span class="fl">0.4</span>, <span class="dt">outlier.shape =</span> <span class="ot">NA</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb69-8" title="8"><span class="st">  </span></a>
<a class="sourceLine" id="cb69-9" title="9"><span class="st">  </span><span class="kw">facet_grid</span>(.<span class="op">~</span>a_Group)</a>
<a class="sourceLine" id="cb69-10" title="10"><span class="co">#&gt; Warning: Removed 34 rows containing missing values (geom_point).</span></a></code></pre></div>
<p><img src="2_data_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p>For comparison of relatedness estimators check Yaka et al. 2021 Curr Biol.</p>
<hr />

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="project-notes.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": "github"
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
